

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>Example Scripts (for Developers) &mdash; TRANSIT v3.3.10 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/default.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="pytransit package" href="transit.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> TRANSIT
          

          
            
            <img src="_static/transit_logo.png" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                v3.3.10
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption" role="heading"><span class="caption-text">TRANSIT MANUAL</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="transit_overview.html">TRANSIT Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_install.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_running.html">Running TRANSIT</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_quality_control.html">Quality Control</a></li>
<li class="toctree-l1"><a class="reference internal" href="file_formats.html">File Formats for Transit</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">TnSeq ANALYSIS METHODS</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="method_normalization.html">Normalization</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_tnseq_stats.html">tnseq_stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_gumbel.html">Gumbel</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_griffin.html">Griffin</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_ttnfitness.html">TTN-Fitness</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_tn5gaps.html">Tn5Gaps</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_HMM.html">HMM</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_resampling.html">Resampling</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_Utest.html">Mann-Whitney U-test (utest)</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_anova.html">ANOVA</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_corrplot.html">corrplot</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_heatmap.html">heatmap</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_GI.html">Genetic Interactions</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_zinb.html">ZINB</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_pathway_enrichment.html">Pathway Enrichment Analysis</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">TRANSIT Tutorials</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="transit_interactions_tutorial.html">Tutorial: Genetic Interactions Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_normalization_tutorial.html">Tutorial: Normalize datasets</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_export_tutorial.html">Tutorial: Export datasets</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_console_cheatsheet.html">Console Mode Cheat-Sheet</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">TPP Manual</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="tpp.html">TPP Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#installation">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#running-tpp">Running TPP</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#mapping-to-genomes-with-multiple-contigs">Mapping to Genomes with Multiple Contigs</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#overview-of-data-processing-procedure">Overview of Data Processing Procedure</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#statistics">Statistics</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Code Documentation</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="transit.html">pytransit package</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Example Scripts (for Developers)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#example-1-src-pytransit-resampling-py">Example 1: src/pytransit_resampling.py</a></li>
<li class="toctree-l2"><a class="reference internal" href="#example-2-src-allpairs-resampling-py">Example 2: src/allpairs_resampling.py</a></li>
</ul>
</li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">TRANSIT</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>Example Scripts (for Developers)</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/example_scripts.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="example-scripts-for-developers">
<h1>Example Scripts (for Developers)<a class="headerlink" href="#example-scripts-for-developers" title="Permalink to this headline">¶</a></h1>
<div class="line-block">
<div class="line"><br /></div>
</div>
<p>Here are two example python scripts for developers showing how to use the <em>pytransit</em> package.</p>
<div class="section" id="example-1-src-pytransit-resampling-py">
<h2>Example 1: src/pytransit_resampling.py<a class="headerlink" href="#example-1-src-pytransit-resampling-py" title="Permalink to this headline">¶</a></h2>
<p>This is a stand-alone script that uses the pytransit library to do
resampling between two groups of wig files, analogous to running
“python3 transit.py resampling…”.  It is a simplified version of
src/pytransit/analysis/resampling.py, which has more options.</p>
<p>To run this, cd into your transit/src/ directory.  It prints output to
stdout. <strong>Be sure to put transit/src/ in your PYTHON_PATH</strong>
(e.g. using export in bash, or setenv in csh).</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">usage</span><span class="p">:</span> <span class="n">python3</span> <span class="n">pytransit_resampling</span><span class="o">.</span><span class="n">py</span> <span class="o">&lt;</span><span class="n">comma_separated_wigs_condA</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">comma_separated_wigs_condB</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">prot_table</span><span class="o">&gt;</span>

<span class="o">&gt;</span> <span class="n">cd</span> <span class="n">transit</span><span class="o">/</span><span class="n">src</span><span class="o">/</span>
<span class="o">&gt;</span> <span class="n">python3</span> <span class="n">pytransit_resampling</span><span class="o">.</span><span class="n">py</span> <span class="n">pytransit</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">glycerol_H37Rv_rep1</span><span class="o">.</span><span class="n">wig</span><span class="p">,</span><span class="n">pytransit</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">glycerol_H37Rv_rep2</span><span class="o">.</span><span class="n">wig</span> <span class="n">pytransit</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">cholesterol_H37Rv_rep1</span><span class="o">.</span><span class="n">wig</span><span class="p">,</span><span class="n">pytransit</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">cholesterol_H37Rv_rep2</span><span class="o">.</span><span class="n">wig</span><span class="p">,</span><span class="n">pytransit</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">cholesterol_H37Rv_rep3</span><span class="o">.</span><span class="n">wig</span> <span class="n">pytransit</span><span class="o">/</span><span class="n">genomes</span><span class="o">/</span><span class="n">H37Rv</span><span class="o">.</span><span class="n">prot_table</span>
</pre></div>
</div>
<p>The important functions illustrated in this script are:</p>
<blockquote>
<div><ul class="simple">
<li><p><strong>transit_tools.get_validated_data()</strong></p></li>
<li><p><strong>norm_tools.normalize_data()</strong></p></li>
<li><p><strong>stat_tools.resampling()</strong></p></li>
</ul>
</div></blockquote>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span> <span class="c1"># transit/src/pytransit_resampling.py</span>
 <span class="kn">import</span> <span class="nn">sys</span><span class="o">,</span><span class="nn">numpy</span>
 <span class="kn">import</span> <span class="nn">pytransit.transit_tools</span> <span class="k">as</span> <span class="nn">transit_tools</span>
 <span class="kn">import</span> <span class="nn">pytransit.tnseq_tools</span> <span class="k">as</span> <span class="nn">tnseq_tools</span>
 <span class="kn">import</span> <span class="nn">pytransit.stat_tools</span> <span class="k">as</span> <span class="nn">stat_tools</span>
 <span class="kn">import</span> <span class="nn">pytransit.norm_tools</span> <span class="k">as</span> <span class="nn">norm_tools</span>
 <span class="kn">from</span> <span class="nn">statsmodels.stats.multitest</span> <span class="kn">import</span> <span class="n">fdrcorrection</span>

 <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">)</span><span class="o">!=</span><span class="mi">4</span><span class="p">:</span>
   <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;usage: python3 pytransit_resampling.py &lt;comma_separated_wigs_condA&gt; &lt;comma_separated_wigs_condB&gt; &lt;prot_table&gt;&quot;</span><span class="p">)</span>
   <span class="n">sys</span><span class="o">.</span><span class="n">exit</span><span class="p">(</span><span class="mi">0</span><span class="p">)</span>

 <span class="n">wigsA</span> <span class="o">=</span> <span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;,&#39;</span><span class="p">)</span>
 <span class="n">wigsB</span> <span class="o">=</span> <span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;,&#39;</span><span class="p">)</span>
 <span class="n">nA</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">wigsA</span><span class="p">)</span>
 <span class="n">prot_table</span> <span class="o">=</span> <span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">[</span><span class="mi">3</span><span class="p">]</span>

 <span class="p">(</span><span class="n">counts</span><span class="p">,</span><span class="n">sites</span><span class="p">)</span> <span class="o">=</span> <span class="n">transit_tools</span><span class="o">.</span><span class="n">get_validated_data</span><span class="p">(</span><span class="n">wigsA</span><span class="o">+</span><span class="n">wigsB</span><span class="p">)</span> <span class="c1"># data is a DxN numpy array, D=num datasets, N=num TA sites</span>
 <span class="nb">print</span><span class="p">(</span><span class="n">counts</span><span class="o">.</span><span class="n">shape</span><span class="p">)</span>

 <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;normalizing data with TTR&quot;</span><span class="p">)</span>
 <span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">factors</span><span class="p">)</span> <span class="o">=</span> <span class="n">norm_tools</span><span class="o">.</span><span class="n">normalize_data</span><span class="p">(</span><span class="n">counts</span><span class="p">,</span> <span class="s2">&quot;TTR&quot;</span><span class="p">)</span>

 <span class="c1"># this divides the raw data (counts at all TA sites) into a smaller array of counts for each gene based on its coordinates</span>
<span class="n">genes</span> <span class="o">=</span> <span class="n">tnseq_tools</span><span class="o">.</span><span class="n">Genes</span><span class="p">(</span><span class="kc">None</span><span class="p">,</span><span class="n">prot_table</span><span class="p">,</span><span class="n">data</span><span class="o">=</span><span class="n">data</span><span class="p">,</span> <span class="n">position</span><span class="o">=</span><span class="n">sites</span><span class="p">)</span>

 <span class="n">results</span><span class="p">,</span><span class="n">pvals</span> <span class="o">=</span> <span class="p">[],[]</span>
 <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">genes</span><span class="p">:</span>
   <span class="n">ii</span> <span class="o">=</span> <span class="n">gene</span><span class="o">.</span><span class="n">reads</span> <span class="c1"># coordinates of TA sites in gene</span>
   <span class="k">if</span> <span class="n">gene</span><span class="o">.</span><span class="n">n</span><span class="o">==</span><span class="mi">0</span><span class="p">:</span> <span class="k">continue</span> <span class="c1"># skip genes with 0 TA sites</span>
   <span class="n">data1</span> <span class="o">=</span> <span class="n">gene</span><span class="o">.</span><span class="n">reads</span><span class="p">[:</span><span class="n">nA</span><span class="p">]</span> <span class="c1"># counts at TA sites in gene for wigs for condition A</span>
   <span class="n">data2</span> <span class="o">=</span> <span class="n">gene</span><span class="o">.</span><span class="n">reads</span><span class="p">[</span><span class="n">nA</span><span class="p">:]</span>
   <span class="n">stats</span> <span class="o">=</span> <span class="n">stat_tools</span><span class="o">.</span><span class="n">resampling</span><span class="p">(</span><span class="n">data1</span><span class="p">,</span> <span class="n">data2</span><span class="p">,</span> <span class="n">adaptive</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span> <span class="c1"># use defaults for other params</span>
   <span class="p">(</span><span class="n">test_obs</span><span class="p">,</span> <span class="n">mean1</span><span class="p">,</span> <span class="n">mean2</span><span class="p">,</span> <span class="n">log2FC</span><span class="p">,</span> <span class="n">pval_ltail</span><span class="p">,</span> <span class="n">pval_utail</span><span class="p">,</span>  <span class="n">pval_2tail</span><span class="p">,</span> <span class="n">testlist</span><span class="p">)</span> <span class="o">=</span> <span class="n">stats</span> <span class="c1">#     unpack</span>
   <span class="n">vals</span> <span class="o">=</span> <span class="p">[</span><span class="n">gene</span><span class="o">.</span><span class="n">orf</span><span class="p">,</span><span class="n">gene</span><span class="o">.</span><span class="n">name</span><span class="p">,</span><span class="n">gene</span><span class="o">.</span><span class="n">n</span><span class="p">,</span><span class="nb">round</span><span class="p">(</span><span class="n">mean1</span><span class="p">,</span><span class="mi">1</span><span class="p">),</span><span class="nb">round</span><span class="p">(</span><span class="n">mean2</span><span class="p">,</span><span class="mi">1</span><span class="p">),</span><span class="nb">round</span><span class="p">(</span><span class="n">log2FC</span><span class="p">,</span><span class="mi">3</span><span class="p">),</span><span class="n">pval_2tail</span><span class="p">]</span>
   <span class="n">results</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">vals</span><span class="p">)</span>

 <span class="c1"># do multiple-tests correction</span>
 <span class="n">pvals</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">results</span><span class="p">]</span>
 <span class="n">qvals</span> <span class="o">=</span> <span class="n">fdrcorrection</span><span class="p">(</span><span class="n">pvals</span><span class="p">)[</span><span class="mi">1</span><span class="p">]</span>

 <span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="s2">&quot;orf gene numTA meanA meanB LFC Pval Qval&quot;</span><span class="o">.</span><span class="n">split</span><span class="p">()))</span>
 <span class="k">for</span> <span class="n">vals</span><span class="p">,</span><span class="n">qval</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">results</span><span class="p">,</span><span class="n">qvals</span><span class="p">):</span>
   <span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">vals</span><span class="o">+</span><span class="p">[</span><span class="n">qval</span><span class="p">]]))</span>
 <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%s</span><span class="s2">/</span><span class="si">%s</span><span class="s2"> hits (qval&lt;0.05)&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="nb">sum</span><span class="p">([</span><span class="n">q</span><span class="o">&lt;</span><span class="mf">0.05</span> <span class="k">for</span> <span class="n">q</span> <span class="ow">in</span> <span class="n">qvals</span><span class="p">]),</span><span class="nb">len</span><span class="p">(</span><span class="n">results</span><span class="p">)))</span>
</pre></div>
</div>
</div>
<div class="section" id="example-2-src-allpairs-resampling-py">
<h2>Example 2: src/allpairs_resampling.py<a class="headerlink" href="#example-2-src-allpairs-resampling-py" title="Permalink to this headline">¶</a></h2>
<p>While the example above shows how to read-in and process individual
wig files, this examples shows how to work with <a class="reference internal" href="file_formats.html#combined-wig"><span class="std std-ref">combined_wig and
sample metadata files</span></a>.  It does resampling between
each pair of conditions, and prints out a matrix of hits
(statistically-signficant conditionally-essential genes).</p>
<p>To run this, cd into your transit/src/ directory.  It prints output to
stdout. <strong>Be sure to put transit/src/ in your PYTHON_PATH</strong>
(e.g. using export in bash, or setenv in csh).</p>
<p>The important parts illustrated in this example script are:</p>
<blockquote>
<div><ul class="simple">
<li><p>the <strong>tnseq_tools.read_combined_wig()</strong> function</p></li>
<li><p>how to read the metadata file</p></li>
<li><p>selecting counts from the data matrix for the samples associated with each condition</p></li>
</ul>
</div></blockquote>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">sys</span><span class="o">,</span><span class="nn">numpy</span>
<span class="kn">import</span> <span class="nn">pytransit.transit_tools</span> <span class="k">as</span> <span class="nn">transit_tools</span>
<span class="kn">import</span> <span class="nn">pytransit.tnseq_tools</span> <span class="k">as</span> <span class="nn">tnseq_tools</span>
<span class="kn">import</span> <span class="nn">pytransit.stat_tools</span> <span class="k">as</span> <span class="nn">stat_tools</span>
<span class="kn">import</span> <span class="nn">pytransit.norm_tools</span> <span class="k">as</span> <span class="nn">norm_tools</span>
<span class="kn">from</span> <span class="nn">statsmodels.stats.multitest</span> <span class="kn">import</span> <span class="n">fdrcorrection</span>

<span class="c1"># this is a stand-alone script that uses the pytransit library to do resampling between all pairs of conditions in a combined_wig file</span>
<span class="c1"># metadata file indicates which replicates belong to which conditions</span>
<span class="c1"># put transit/src/ in your PYTHON_PATH (e.g. using export in bash, or setenv in csh)</span>
<span class="c1"># prints output to stdout</span>

<span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">)</span><span class="o">!=</span><span class="mi">4</span><span class="p">:</span>
  <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;usage: python3 allpairs_resampling.py &lt;combined_wig_file&gt; &lt;metadata_file&gt; &lt;prot_table&gt;&quot;</span><span class="p">)</span>
  <span class="n">sys</span><span class="o">.</span><span class="n">exit</span><span class="p">(</span><span class="mi">0</span><span class="p">)</span>

<span class="n">combined_wig_file</span> <span class="o">=</span> <span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
<span class="n">metadata_file</span> <span class="o">=</span> <span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span>
<span class="n">prot_table</span> <span class="o">=</span> <span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">[</span><span class="mi">3</span><span class="p">]</span>

<span class="c1">#################################</span>

<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;reading data&quot;</span><span class="p">)</span>
<span class="p">(</span><span class="n">sites</span><span class="p">,</span> <span class="n">data</span><span class="p">,</span> <span class="n">filenamesInCombWig</span><span class="p">)</span> <span class="o">=</span> <span class="n">tnseq_tools</span><span class="o">.</span><span class="n">read_combined_wig</span><span class="p">(</span><span class="n">combined_wig_file</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;data.shape =&quot;</span><span class="p">,</span><span class="n">data</span><span class="o">.</span><span class="n">shape</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;normalizing using TTR&quot;</span><span class="p">)</span>
<span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">factors</span><span class="p">)</span> <span class="o">=</span> <span class="n">norm_tools</span><span class="o">.</span><span class="n">normalize_data</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="s2">&quot;TTR&quot;</span><span class="p">)</span>

<span class="c1"># there is a tnseq_tools.read_samples_metadata(), but it is kind of complicated</span>
<span class="c1"># so just read through tab-separated file and collect sample Filenames associated with Conditions</span>
<span class="c1"># metadata files have at least 3 columns: Id, Filename, Condition</span>
<span class="c1"># the columns in the combined_wig are cross-referenced by Filename (from the original wigs)</span>

<span class="n">Conditions</span><span class="p">,</span><span class="n">Samples</span> <span class="o">=</span> <span class="p">[],{}</span>
<span class="n">CondCol</span><span class="p">,</span><span class="n">FnameCol</span> <span class="o">=</span> <span class="o">-</span><span class="mi">1</span><span class="p">,</span><span class="o">-</span><span class="mi">1</span>
<span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="nb">open</span><span class="p">(</span><span class="n">metadata_file</span><span class="p">):</span>
  <span class="n">w</span> <span class="o">=</span> <span class="n">line</span><span class="o">.</span><span class="n">rstrip</span><span class="p">()</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span> <span class="c1"># tab-separated</span>
  <span class="k">if</span> <span class="n">CondCol</span><span class="o">==-</span><span class="mi">1</span><span class="p">:</span> <span class="n">CondCol</span><span class="p">,</span><span class="n">FnameCol</span> <span class="o">=</span> <span class="n">w</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s2">&quot;Condition&quot;</span><span class="p">),</span><span class="n">w</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s2">&quot;Filename&quot;</span><span class="p">);</span> <span class="k">continue</span> <span class="c1"># error if headers not found</span>
  <span class="n">cond</span><span class="p">,</span><span class="n">fname</span> <span class="o">=</span> <span class="n">w</span><span class="p">[</span><span class="n">CondCol</span><span class="p">],</span><span class="n">w</span><span class="p">[</span><span class="n">FnameCol</span><span class="p">]</span>
  <span class="k">if</span> <span class="n">cond</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">Conditions</span><span class="p">:</span> <span class="n">Conditions</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">cond</span><span class="p">)</span>
  <span class="k">if</span> <span class="n">cond</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">Samples</span><span class="p">:</span> <span class="n">Samples</span><span class="p">[</span><span class="n">cond</span><span class="p">]</span> <span class="o">=</span> <span class="p">[]</span>
  <span class="n">Samples</span><span class="p">[</span><span class="n">cond</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">Conditions</span><span class="se">\t</span><span class="s2">: Samples&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;-------------------------&quot;</span><span class="p">)</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span><span class="n">cond</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">Conditions</span><span class="p">):</span>
  <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%s</span><span class="s2">:</span><span class="si">%-8s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">i</span><span class="o">+</span><span class="mi">1</span><span class="p">,</span><span class="n">cond</span><span class="p">),</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">: &quot;</span><span class="p">,</span><span class="s2">&quot;, &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">Samples</span><span class="p">[</span><span class="n">cond</span><span class="p">]))</span>

<span class="c1">#################################</span>

<span class="n">genes</span> <span class="o">=</span> <span class="n">tnseq_tools</span><span class="o">.</span><span class="n">Genes</span><span class="p">(</span><span class="kc">None</span><span class="p">,</span><span class="n">prot_table</span><span class="p">,</span><span class="n">data</span><span class="o">=</span><span class="n">data</span><span class="p">,</span> <span class="n">position</span><span class="o">=</span><span class="n">sites</span><span class="p">)</span> <span class="c1"># divides counts at all TAs sites into groups by orf</span>

<span class="nb">print</span><span class="p">()</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;running resampling on each pair of conditions...&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;reporting number of conditionally essential genes (Qval&lt;0.05)&quot;</span><span class="p">)</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span><span class="n">condA</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">Conditions</span><span class="p">):</span>
  <span class="k">for</span> <span class="n">j</span><span class="p">,</span><span class="n">condB</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">Conditions</span><span class="p">):</span>
    <span class="k">if</span> <span class="n">i</span><span class="o">&lt;</span><span class="n">j</span><span class="p">:</span>
      <span class="n">pvals</span> <span class="o">=</span> <span class="p">[]</span>
      <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">genes</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">gene</span><span class="o">.</span><span class="n">n</span><span class="o">==</span><span class="mi">0</span><span class="p">:</span> <span class="k">continue</span> <span class="c1"># skip genes with 0 TA sites</span>
        <span class="n">idxA</span> <span class="o">=</span> <span class="p">[</span><span class="n">filenamesInCombWig</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">s</span><span class="p">)</span> <span class="k">for</span> <span class="n">s</span> <span class="ow">in</span> <span class="n">Samples</span><span class="p">[</span><span class="n">condA</span><span class="p">]]</span>
        <span class="n">idxB</span> <span class="o">=</span> <span class="p">[</span><span class="n">filenamesInCombWig</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">s</span><span class="p">)</span> <span class="k">for</span> <span class="n">s</span> <span class="ow">in</span> <span class="n">Samples</span><span class="p">[</span><span class="n">condB</span><span class="p">]]</span>
        <span class="n">data1</span> <span class="o">=</span> <span class="n">gene</span><span class="o">.</span><span class="n">reads</span><span class="p">[</span><span class="n">idxA</span><span class="p">]</span> <span class="c1"># counts at TA sites in gene for wigs for condition A</span>
        <span class="n">data2</span> <span class="o">=</span> <span class="n">gene</span><span class="o">.</span><span class="n">reads</span><span class="p">[</span><span class="n">idxB</span><span class="p">]</span>
        <span class="n">stats</span> <span class="o">=</span> <span class="n">stat_tools</span><span class="o">.</span><span class="n">resampling</span><span class="p">(</span><span class="n">data1</span><span class="p">,</span> <span class="n">data2</span><span class="p">,</span> <span class="n">adaptive</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span> <span class="c1"># use defaults for other params</span>
        <span class="p">(</span><span class="n">test_obs</span><span class="p">,</span> <span class="n">mean1</span><span class="p">,</span> <span class="n">mean2</span><span class="p">,</span> <span class="n">log2FC</span><span class="p">,</span> <span class="n">pval_ltail</span><span class="p">,</span> <span class="n">pval_utail</span><span class="p">,</span>  <span class="n">pval_2tail</span><span class="p">,</span> <span class="n">testlist</span><span class="p">)</span> <span class="o">=</span> <span class="n">stats</span> <span class="c1"># unpack</span>
        <span class="n">pvals</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">pval_2tail</span><span class="p">)</span>

      <span class="n">qvals</span> <span class="o">=</span> <span class="n">fdrcorrection</span><span class="p">(</span><span class="n">pvals</span><span class="p">)[</span><span class="mi">1</span><span class="p">]</span>
      <span class="n">numhits</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">([</span><span class="n">q</span><span class="o">&lt;</span><span class="mf">0.05</span> <span class="k">for</span> <span class="n">q</span> <span class="ow">in</span> <span class="n">qvals</span><span class="p">])</span>
      <span class="n">vals</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;</span><span class="si">%s</span><span class="s2">:</span><span class="si">%-8s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">i</span><span class="o">+</span><span class="mi">1</span><span class="p">,</span><span class="n">condA</span><span class="p">),</span><span class="s2">&quot;</span><span class="si">%s</span><span class="s2">:</span><span class="si">%-8s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">j</span><span class="o">+</span><span class="mi">1</span><span class="p">,</span><span class="n">condB</span><span class="p">),</span><span class="n">numhits</span><span class="p">]</span>
      <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">vals</span><span class="p">]))</span>
</pre></div>
</div>
<p>Here is the output of this script for data from growth
of M. tuberculosis H37Rv on media containing iron supplied by
different vehicles (e.g. mycobactin, carboxymycobactin, hemin,
hemoglobin…), which requires genes in different pathways for uptake
<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/32069330/">(Zhang et al., 2020)</a>.
The raw data (wig files, with insertion counts at TA sites) have been
combined into a <strong>combined_wig file</strong> and a <strong>metatdata file</strong> that
describes which samples belong to which condition.  These files
(<em>iron_combined_wig4.txt</em> and <em>iron_samples_metadata.txt</em>) can be found in
the transit data directory, transit/src/pytransit/data/.  You can
also compare this to the heatmap shown on the page for <a class="reference internal" href="method_corrplot.html#corrplot"><span class="std std-ref">corrplot</span></a>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">cd</span> <span class="n">transit</span><span class="o">/</span><span class="n">src</span><span class="o">/</span>
<span class="o">&gt;</span> <span class="n">python3</span> <span class="n">allpairs_resampling</span><span class="o">.</span><span class="n">py</span> <span class="n">pytransit</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">iron_combined_wig4</span><span class="o">.</span><span class="n">txt</span> <span class="n">pytransit</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">iron_samples_metadata</span><span class="o">.</span><span class="n">txt</span> <span class="n">pytransit</span><span class="o">/</span><span class="n">genomes</span><span class="o">/</span><span class="n">H37Rv</span><span class="o">.</span><span class="n">prot_table</span>
<span class="n">reading</span> <span class="n">data</span>
<span class="n">data</span><span class="o">.</span><span class="n">shape</span> <span class="o">=</span> <span class="p">(</span><span class="mi">14</span><span class="p">,</span> <span class="mi">74605</span><span class="p">)</span>
<span class="n">normalizing</span> <span class="n">using</span> <span class="n">TTR</span>

<span class="n">Conditions</span>    <span class="p">:</span> <span class="n">Samples</span>
<span class="o">-------------------------</span>
<span class="mi">1</span><span class="p">:</span><span class="n">HighFeMBT</span>   <span class="p">:</span>  <span class="n">HighFeMBT</span><span class="o">.</span><span class="n">wig</span><span class="p">,</span> <span class="n">HighFeMBT2</span><span class="o">.</span><span class="n">wig</span><span class="p">,</span> <span class="n">HighFeMBT3</span><span class="o">.</span><span class="n">wig</span>
<span class="mi">2</span><span class="p">:</span><span class="n">LowFeMBT</span>    <span class="p">:</span>  <span class="n">LowFeMBT</span><span class="o">.</span><span class="n">wig</span><span class="p">,</span> <span class="n">LowFeMBT2</span><span class="o">.</span><span class="n">wig</span>
<span class="mi">3</span><span class="p">:</span><span class="n">FeCMBT</span>      <span class="p">:</span>  <span class="n">FeCMBT</span><span class="o">.</span><span class="n">wig</span><span class="p">,</span> <span class="n">FeCMBT2b</span><span class="o">.</span><span class="n">wig</span>
<span class="mi">4</span><span class="p">:</span><span class="n">Hemin</span>       <span class="p">:</span>  <span class="n">Hemin</span><span class="o">.</span><span class="n">wig</span><span class="p">,</span> <span class="n">Hemin2b</span><span class="o">.</span><span class="n">wig</span><span class="p">,</span> <span class="n">Hemin3b</span><span class="o">.</span><span class="n">wig</span>
<span class="mi">5</span><span class="p">:</span><span class="n">Hemoglobin</span>  <span class="p">:</span>  <span class="n">Hemoglobin</span><span class="o">.</span><span class="n">wig</span><span class="p">,</span> <span class="n">Hemoglobin2b</span><span class="o">.</span><span class="n">wig</span>
<span class="mi">6</span><span class="p">:</span><span class="n">HeminMBT</span>    <span class="p">:</span>  <span class="n">HeminMBT</span><span class="o">.</span><span class="n">wig</span><span class="p">,</span> <span class="n">HeminMBT2</span><span class="o">.</span><span class="n">wig</span>

<span class="n">running</span> <span class="n">resampling</span> <span class="n">on</span> <span class="n">each</span> <span class="n">pair</span> <span class="n">of</span> <span class="n">conditions</span><span class="o">...</span>
<span class="n">reporting</span> <span class="n">number</span> <span class="n">of</span> <span class="n">conditionally</span> <span class="n">essential</span> <span class="n">genes</span> <span class="p">(</span><span class="n">Qval</span><span class="o">&lt;</span><span class="mf">0.05</span><span class="p">)</span>
<span class="mi">1</span><span class="p">:</span><span class="n">HighFeMBT</span>   <span class="mi">2</span><span class="p">:</span><span class="n">LowFeMBT</span>      <span class="mi">38</span>
<span class="mi">1</span><span class="p">:</span><span class="n">HighFeMBT</span>   <span class="mi">3</span><span class="p">:</span><span class="n">FeCMBT</span>        <span class="mi">10</span>
<span class="mi">1</span><span class="p">:</span><span class="n">HighFeMBT</span>   <span class="mi">4</span><span class="p">:</span><span class="n">Hemin</span>         <span class="mi">136</span>
<span class="mi">1</span><span class="p">:</span><span class="n">HighFeMBT</span>   <span class="mi">5</span><span class="p">:</span><span class="n">Hemoglobin</span>    <span class="mi">134</span>
<span class="mi">1</span><span class="p">:</span><span class="n">HighFeMBT</span>   <span class="mi">6</span><span class="p">:</span><span class="n">HeminMBT</span>      <span class="mi">30</span>
<span class="mi">2</span><span class="p">:</span><span class="n">LowFeMBT</span>    <span class="mi">3</span><span class="p">:</span><span class="n">FeCMBT</span>        <span class="mi">35</span>
<span class="mi">2</span><span class="p">:</span><span class="n">LowFeMBT</span>    <span class="mi">4</span><span class="p">:</span><span class="n">Hemin</span>         <span class="mi">70</span>
<span class="mi">2</span><span class="p">:</span><span class="n">LowFeMBT</span>    <span class="mi">5</span><span class="p">:</span><span class="n">Hemoglobin</span>    <span class="mi">71</span>
<span class="mi">2</span><span class="p">:</span><span class="n">LowFeMBT</span>    <span class="mi">6</span><span class="p">:</span><span class="n">HeminMBT</span>      <span class="mi">5</span>
<span class="mi">3</span><span class="p">:</span><span class="n">FeCMBT</span>      <span class="mi">4</span><span class="p">:</span><span class="n">Hemin</span>         <span class="mi">104</span>
<span class="mi">3</span><span class="p">:</span><span class="n">FeCMBT</span>      <span class="mi">5</span><span class="p">:</span><span class="n">Hemoglobin</span>    <span class="mi">106</span>
<span class="mi">3</span><span class="p">:</span><span class="n">FeCMBT</span>      <span class="mi">6</span><span class="p">:</span><span class="n">HeminMBT</span>      <span class="mi">44</span>
<span class="mi">4</span><span class="p">:</span><span class="n">Hemin</span>       <span class="mi">5</span><span class="p">:</span><span class="n">Hemoglobin</span>    <span class="mi">4</span>
<span class="mi">4</span><span class="p">:</span><span class="n">Hemin</span>       <span class="mi">6</span><span class="p">:</span><span class="n">HeminMBT</span>      <span class="mi">77</span>
<span class="mi">5</span><span class="p">:</span><span class="n">Hemoglobin</span>  <span class="mi">6</span><span class="p">:</span><span class="n">HeminMBT</span>      <span class="mi">89</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Note, in allpairs_resampling.py, the FDR correction
to adjust P-values for testing all genes in parallel
is applied within each pairwise comparison.
This correction is then repeated independently for each pair
of conditions analyzed.  Formally, it would be more rigorous to apply
the FDR correction one time at the end, to adjust P-values over all pairs and
over all genes (which would be G*N*(N-1)/2 probabilities, where G in
the number of genes in the genome, and N is the number of conditions).
This would likely further reduce the number of significant
conditionally-essential genes.  But for simplicity, this example
script does not do that, because it is just designed to illustrate
using the <em>pytransit</em> package.</p>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="transit.html" class="btn btn-neutral float-left" title="pytransit package" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2015, Michael A. DeJesus &amp; Thomas R. Ioerger.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>