

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>TRANSIT Overview (4.0) &mdash; TRANSIT v4.0.0 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/default.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Installation" href="transit_install.html" />
    <link rel="prev" title="TRANSIT Overview" href="transit_overview.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> TRANSIT
          

          
            
            <img src="_static/transit_logo.png" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                v4.0.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption" role="heading"><span class="caption-text">TRANSIT MANUAL</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="transit_overview.html">TRANSIT Overview</a><ul>
<li class="toctree-l2"><a class="reference internal" href="transit_overview.html#developers">Developers</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_overview.html#references">References</a></li>
</ul>
</li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">TRANSIT Overview (4.0)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#typical-workflow">Typical Workflow</a></li>
<li class="toctree-l2"><a class="reference internal" href="#tpp-transit-pre-processor">TPP (Transit Pre-Processor)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#combined-wig-and-metadata-files">Combined_wig and Metadata Files</a></li>
<li class="toctree-l2"><a class="reference internal" href="#genome-annotations-prot-tables-and-gff-files">Genome Annotations (.prot_tables and .gff files)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#gui">GUI</a></li>
<li class="toctree-l2"><a class="reference internal" href="#pre-processing-tasks">Pre-Processing Tasks</a></li>
<li class="toctree-l2"><a class="reference internal" href="#statistical-analysis-methods">Statistical Analysis Methods</a></li>
<li class="toctree-l2"><a class="reference internal" href="#analyses-for-tn5-datasets">Analyses for Tn5 Datasets</a></li>
<li class="toctree-l2"><a class="reference internal" href="#results-and-post-processing">Results and Post-Processing</a></li>
<li class="toctree-l2"><a class="reference internal" href="#command-line">Command Line</a></li>
<li class="toctree-l2"><a class="reference internal" href="#developers">Developers</a></li>
<li class="toctree-l2"><a class="reference internal" href="#references">References</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="transit_install.html">Installation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="transit_install.html#requirements">Requirements</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_install.html#python-2-7">Python 2.7:</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_install.html#python-3">Python 3:</a><ul>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#additional-requirements-r-statistical-analysis-package">Additional Requirements: R (statistical analysis package)</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="transit_install.html#use-as-a-python-package">Use as a Python Package</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_install.html#optional-install-bwa-to-use-with-tpp-pre-processor">Optional: Install BWA to use with TPP pre-processor</a><ul>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#linux-osx-instructions">Linux &amp; OSX Instructions</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#windows-instructions">Windows Instructions</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="transit_install.html#upgrading">Upgrading</a><ul>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#method-1-upgrading-package-installation">Method 1: Upgrading package installation</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#method-2-upgrading-source-installation">Method 2: Upgrading source installation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="transit_install.html#installing-wxpython">Installing wxPython</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_install.html#troubleshooting">Troubleshooting</a><ul>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#no-window-appears-when-running-in-gui-mode">1. No window appears when running in GUI mode.</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#pip-systemerror-cannot-compile-python-h">2. pip: SystemError: Cannot compile ‘Python.h’.</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#pip-the-following-required-packages-can-not-be-built-freetype-png-etc">3. pip: “The following required packages can not be built: freetype,png,” etc.</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#pip-no-lapack-blas-resources-found">4. pip: “No lapack/blas resources found”</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#resources-contextualversionconflict-six-1-5-2">5. “resources.ContextualVersionConflict (six 1.5.2)…”</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="transit_running.html">Running TRANSIT</a><ul>
<li class="toctree-l2"><a class="reference internal" href="transit_running.html#gui-mode">GUI Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_running.html#command-line-mode">Command line Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_running.html#prot-tables-annotations">Prot_tables (Annotations)</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_running.html#tn5-datasets">Tn5 Datasets</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="input_files.html">Input Files for Transit</a><ul>
<li class="toctree-l2"><a class="reference internal" href="input_files.html#combined-wig-files">Combined_wig Files</a></li>
<li class="toctree-l2"><a class="reference internal" href="input_files.html#samples-metadata-file">Samples Metadata File</a></li>
<li class="toctree-l2"><a class="reference internal" href="input_files.html#genome-annotations-prot-tables-and-gff-files">Genome Annotations (.prot_tables and .gff files)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="transit_features.html">Quality Control</a><ul>
<li class="toctree-l2"><a class="reference internal" href="transit_features.html#qc-metrics-table">QC Metrics Table</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_features.html#qc-plots">QC Plots</a><ul>
<li class="toctree-l3"><a class="reference internal" href="transit_features.html#figure-1-read-count-distribution">Figure 1: Read-Count Distribution</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_features.html#figure-2-qq-plot-of-read-counts-vs-geometric-distribution">Figure 2: QQ-Plot of Read-Counts vs Geometric Distribution</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_features.html#figure-3-ranked-plot-of-read-counts">Figure 3: Ranked plot of Read-Counts</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="transit_features.html#interpretation-of-data-quality">Interpretation of Data Quality</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_features.html#what-to-do-about-skewed-samples">What to do about Skewed Samples?</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_features.html#beta-geometric-correction">Beta-Geometric Correction</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">PRE-PROCESSING</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="method_tnseq_stats.html">tnseq_stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_scatterplot.html">scatterplot</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_corrplot.html">Corrplot</a><ul>
<li class="toctree-l2"><a class="reference internal" href="method_corrplot.html#command-line-usage">Command Line Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_corrplot.html#gui-mode">GUI Mode</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="method_normalization.html">Normalization</a><ul>
<li class="toctree-l2"><a class="reference internal" href="method_normalization.html#command-line">Command-line</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_normalization.html#example">Example</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="method_gene_means.html">Gene Means</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">ANALYSES of single conditions</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="method_gumbel.html">Gumbel</a><ul>
<li class="toctree-l2"><a class="reference internal" href="method_gumbel.html#how-does-it-work">How does it work?</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_gumbel.html#usage">Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_gumbel.html#parameters">Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_gumbel.html#gui-mode">GUI Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_gumbel.html#outputs-and-diagnostics">Outputs and diagnostics</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_gumbel.html#run-time">Run-time</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="method_HMM.html">HMM</a><ul>
<li class="toctree-l2"><a class="reference internal" href="method_HMM.html#how-does-it-work">How does it work?</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_HMM.html#example">Example</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_HMM.html#parameters">Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_HMM.html#gui-mode">GUI Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_HMM.html#run-time">Run-time</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="method_ttnfitness.html">TTN-Fitness</a><ul>
<li class="toctree-l2"><a class="reference internal" href="method_ttnfitness.html#usage">Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_ttnfitness.html#output">Output</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_ttnfitness.html#example-of-running-the-ttn-fitness-methodology-on-the-sample-glycerol-data">Example of running the TTN-Fitness methodology on the sample glycerol data</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_ttnfitness.html#gui-mode">GUI Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_ttnfitness.html#volcano-plot">Volcano Plot</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">ANALYSES for pairwise comparisons</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="method_resampling.html">Resampling</a><ul>
<li class="toctree-l2"><a class="reference internal" href="method_resampling.html#how-does-it-work">How does it work?</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_resampling.html#site-restricted-s-r-resampling">Site-Restricted (S-R) Resampling</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_resampling.html#usage">Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_resampling.html#parameters">Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_resampling.html#notes">Notes</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_resampling.html#doing-resampling-with-a-combined-wig-file">Doing resampling with a combined_wig file</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_resampling.html#doing-resampling-with-datasets-from-different-libraries">Doing resampling with datasets from different libraries.</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_resampling.html#doing-resampling-between-different-strains">Doing resampling between different strains.</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_resampling.html#output-and-diagnostics">Output and Diagnostics</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_resampling.html#run-time">Run-time</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="method_Utest.html">Mann-Whitney U-test (utest)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="method_Utest.html#usage">Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_Utest.html#output">Output</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_Utest.html#runtime">Runtime</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">ANALYSES for multiple conditions</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="method_GI.html">Genetic Interactions</a><ul>
<li class="toctree-l2"><a class="reference internal" href="method_GI.html#how-does-it-work">How does it work?</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_GI.html#statistical-significance">Statistical Significance</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_GI.html#usage">Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_GI.html#example">Example</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_GI.html#parameters">Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_GI.html#output-and-diagnostics">Output and Diagnostics</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="method_anova.html">ANOVA</a><ul>
<li class="toctree-l2"><a class="reference internal" href="method_anova.html#how-does-it-work">How does it work?</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_anova.html#example">Example</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_anova.html#input-files">Input files</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_anova.html#parameters">Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_anova.html#output-and-diagnostics">Output and Diagnostics</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_anova.html#gui-mode">GUI Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_anova.html#run-time">Run-time</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="method_zinb.html">ZINB</a><ul>
<li class="toctree-l2"><a class="reference internal" href="method_zinb.html#how-does-it-work">How does it work?</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_zinb.html#example">Example</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_zinb.html#input-files">Input files</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_zinb.html#parameters">Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_zinb.html#covariates-and-interactions">Covariates and Interactions</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_zinb.html#categorical-vs-numeric-covariates">Categorical vs Numeric Covariates</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_zinb.html#statistical-significance-what-the-p-values-mean-in-the-zinb-output">Statistical Significance - What the P-values Mean in the ZINB Output</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_zinb.html#output-and-diagnostics">Output and Diagnostics</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_zinb.html#run-time">Run-time</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">POST-PROCESSING</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="method_pathway_enrichment.html">Pathway Enrichment Analysis</a><ul>
<li class="toctree-l2"><a class="reference internal" href="method_pathway_enrichment.html#command-line-usage">Command Line Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_pathway_enrichment.html#parameters">Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_pathway_enrichment.html#auxilliary-pathway-files-in-transit-data-directory">Auxilliary Pathway Files in Transit Data Directory</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_pathway_enrichment.html#current-recommendations">Current Recommendations</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_pathway_enrichment.html#examples">Examples</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_pathway_enrichment.html#interpreting-output-of-pathway-enrichment">Interpreting Output of Pathway Enrichment</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_pathway_enrichment.html#gui-mode">GUI Mode</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="method_heatmap.html">Heatmap</a><ul>
<li class="toctree-l2"><a class="reference internal" href="method_heatmap.html#command-line-usage">Command Line Usage:</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_heatmap.html#gui">GUI:</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Tn5 METHODS</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="method_tn5gaps.html">Tn5Gaps</a><ul>
<li class="toctree-l2"><a class="reference internal" href="method_tn5gaps.html#how-does-it-work">How does it work?</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_tn5gaps.html#usage">Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_tn5gaps.html#parameters">Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_tn5gaps.html#example">Example</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_tn5gaps.html#outputs-and-diagnostics">Outputs and diagnostics</a></li>
<li class="toctree-l2"><a class="reference internal" href="method_tn5gaps.html#run-time">Run-time</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">TRANSIT Tutorials</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="transit_interactions_tutorial.html">Tutorial: Genetic Interactions Analysis</a><ul>
<li class="toctree-l2"><a class="reference internal" href="transit_interactions_tutorial.html#console-mode-tutorial">Console Mode Tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_interactions_tutorial.html#gui-mode-tutorial">GUI Mode Tutorial</a><ul>
<li class="toctree-l3"><a class="reference internal" href="transit_interactions_tutorial.html#run-transit">Run TRANSIT</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_interactions_tutorial.html#adding-the-annotation-file">Adding the annotation file</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_interactions_tutorial.html#adding-datasets-grown-under-condition-a">Adding datasets grown under <em>condition A</em></a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_interactions_tutorial.html#running-the-genetic-interactions-method">Running the Genetic Interactions method</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_interactions_tutorial.html#viewing-gi-results">Viewing GI results</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="transit_normalization_tutorial.html">Tutorial: Normalize datasets</a><ul>
<li class="toctree-l2"><a class="reference internal" href="transit_normalization_tutorial.html#adding-the-annotation-file">Adding the annotation file</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_normalization_tutorial.html#add-wig-datasets">Add .wig datasets</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_normalization_tutorial.html#normalize-and-save">Normalize and Save</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_normalization_tutorial.html#normalization">Normalization</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="transit_export_tutorial.html">Tutorial: Export datasets</a><ul>
<li class="toctree-l2"><a class="reference internal" href="transit_export_tutorial.html#adding-the-annotation-file">Adding the annotation file</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_export_tutorial.html#add-wig-datasets">Add .wig datasets</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_export_tutorial.html#export-to-igv">Export to IGV</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="transit_console_cheatsheet.html">Console Mode Cheat-Sheet</a><ul>
<li class="toctree-l2"><a class="reference internal" href="transit_console_cheatsheet.html#analysis-methods">Analysis Methods</a><ul>
<li class="toctree-l3"><a class="reference internal" href="transit_console_cheatsheet.html#single-condition-essentiality">Single Condition Essentiality</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_console_cheatsheet.html#conditional-essentiality">Conditional Essentiality</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="transit_console_cheatsheet.html#normalizing-datasets">Normalizing datasets</a><ul>
<li class="toctree-l3"><a class="reference internal" href="transit_console_cheatsheet.html#positional-arguments">Positional Arguments</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_console_cheatsheet.html#optional-arguments">Optional Arguments</a></li>
</ul>
</li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">TPP Manual</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="tpp.html">TPP Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#installation">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#running-tpp">Running TPP</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#mapping-to-genomes-with-multiple-contigs">Mapping to Genomes with Multiple Contigs</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#overview-of-data-processing-procedure">Overview of Data Processing Procedure</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#statistics">Statistics</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Code Documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="transit.html">transit package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="transit.html#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit.html#pytransit-norm-tools-module">pytransit.norm_tools module</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit.html#pytransit-stat-tools-module">pytransit.stat_tools module</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit.html#pytransit-tnseq-tools-module">pytransit.tnseq_tools module</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit.html#pytransit-transit-tools-module">pytransit.transit_tools module</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit.html#module-pytransit">Module contents</a></li>
</ul>
</li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">TRANSIT</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>TRANSIT Overview (4.0)</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/transit_overview2.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="transit-overview-4-0">
<h1>TRANSIT Overview (4.0)<a class="headerlink" href="#transit-overview-4-0" title="Permalink to this headline">¶</a></h1>
<p>Transit is a python-based software system that combines statistical
algorithms for analyzing TnSeq data (from sequencing transposon-mutant
libraries).  It has GUI (graphical-user interface) to make it easy for
users to process their data, conduct essentiality analyses, and visualize results.
Transit routines can also be run from the command line.  It works in
Linux, MacOS, and Windows.  Transit requires various python
packages to be installed, as well as R (a statistical program that Transit calls for
some analyses).</p>
<p>Transit has two main phases:</p>
<blockquote>
<div><ul class="simple">
<li><p>a <strong>pre-processing phase</strong> (called TPP) which analyzes raw sequencing data (fastq files) and extracts transposon insertion counts at genomic coordinates (i.e. TA dinucleotides, for the Himar1 transposon), and</p></li>
<li><p><strong>statistical analysis methods</strong> (using Transit GUI or command line) - gumbel, HMM, resampling, ANOVA…</p></li>
</ul>
</div></blockquote>
<p>The statistical analyses are focused on identifying genes
that are <em>essential</em> or <em>conditionally-essential</em>, and quantifying the
statistical significance of these.  Although Transit is primarily
designed for Himar1 transposon libraries, some of the methods can also
be applied to other transposons, like Tn5.</p>
<p><strong>NEW in Transit 4.0:</strong>
The GUI and command-line versions of Transit have been
overhauled to now take <strong>combined_wig</strong> files (and corresponding <strong>metadata</strong> files),
instead of individual .wig files.
Thus, after running TPP on all the individual samples,
the multiple .wig file are combined together in a single combined_wig file
with multiple columns.  This makes it much more convenient to
manage and analyze large datasets.</p>
<div class="section" id="typical-workflow">
<h2>Typical Workflow<a class="headerlink" href="#typical-workflow" title="Permalink to this headline">¶</a></h2>
<p>A typical workflow follows 3 phases, illustrated below.</p>
<ol class="arabic simple">
<li><p>First, the user runs TPP on raw sequence files (.fastq) to produce .wig files with insertion counts at coordinates of TA sites for each sample.</p></li>
<li><p>Second, the user runs a command to merge the .wig files into a single combined_wig file, and prepares a corresponding metadata file describing the samples.</p></li>
<li><p>Third, the user runs various statistical analysis methods in Transit, using the combined_wig and metadata files as inputs.</p></li>
</ol>
<p>Most of the analyses in Transit produce an <strong>output file in tab-separated format that can be
opened as a spreadsheet</strong>.  For example, ‘resampling’ generates an output file
that lists the data (mean insertion counts in 2 conditions being compared) and statistics (log-fold-change, p-value, adjusted p-value)
for each gene in the genome.  The user can then view this output file (in the GUI, or open it in a program like Excel)
to examine genes predicted to be conditionally essential.</p>
<a class="reference internal image-reference" href="_images/Transit_flow_chart2b.png"><img alt="_images/Transit_flow_chart2b.png" class="align-center" src="_images/Transit_flow_chart2b.png" style="width: 600px;" /></a>
</div>
<div class="section" id="tpp-transit-pre-processor">
<h2>TPP (Transit Pre-Processor)<a class="headerlink" href="#tpp-transit-pre-processor" title="Permalink to this headline">¶</a></h2>
<p>TPP is a tool for analyzing raw sequencing data (.fastq files)
and tabulating insertion counts at TA sites.
TPP generates .wig files as output, which
simply contain a pair of columns: 1) coordinates of TA sites,
and 2) counts of insertions observed at each site.</p>
<p>If Himar1 Tn libraries are prepared according to standard protocols,
such as <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/25636614/">(Long et al, 2015)</a>,
then the sequencing reads will have a prefix
correspond to the transposon terminus, followed by genomic DNA (along with template barcodes in read 2).
(See XXXXX for recommendations on sequencing parameters (length, depth), also replicates.)
TPP removes the prefix and maps the genomic portion of the read into
the genome (using BWA, <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/19451168/">Li and Durbin, 2009</a>).
It also reduces counts to unqiue
templates at each site, which reduces noise.</p>
<p>For diagnostics, each TPP run also generates a .tnseq_stats output file
which contains various metrics, such as the density (percent of TA sites
with insertions), total reads, mapped reads, and the NZmean (mean counts at non-zero sites).
It also reports statistics on how many reads were rejected because they contained
primer, adapter, or vectors sequences, which can be useful for diagnosing problems with the library.</p>
<p>For Tn5 datasets (e.g. generated the MmeI restriction enzyme,
<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4696536/">van Opijnen et al, (2015)</a>),
the differences in running TPP are:
1) there is no prefix, since the transposon terminus does not appear in the reads,
and 2) the .wig files contain coordinates and counts for all sites in the
genome, since Tn5 is not restricted to insertions at TA sites (unlike Himar1).</p>
</div>
<div class="section" id="combined-wig-and-metadata-files">
<h2>Combined_wig and Metadata Files<a class="headerlink" href="#combined-wig-and-metadata-files" title="Permalink to this headline">¶</a></h2>
<p>Once the .wig files are created, they are combined into a <a class="reference internal" href="input_files.html#combined-wig-link"><span class="std std-ref">combined_wig</span></a>
file.  This is a new step in Transit 4.0.  Multiple .wig files are
combined into a single combined_wig file using the <strong>‘transit export
combined_wig’</strong> command on the command line.  Users then prepare a
<a class="reference internal" href="input_files.html#metadata-files"><span class="std std-ref">metadata</span></a> file describing each of the samples (as a spreadsheet,
e.g. in Excel, which is then saved in tab-separted format).  Most
commonly, there will be several replicates associated with
each condition.  The analysis methods in Transit all have been redesigned
to take combined_wig and metadata files as inputs, which simplifies
things when working with large datasets.</p>
<p>By default, the insertion counts in each dataset (.wig file) are <strong>normalized</strong>
by TTR (Total Trimmed Read-count) when they are combined in a combined_wig file.
This facilitates comparing insertion counts at individaul TA sites across samples.
If alternative normalization (or none) is desired, this can be specified
using a flag in the ‘export combined_wig’ command.</p>
<p>It is often useful to <em>examine at the pattern of insertions in conditionally-essential genes
in combined_wig files</em> to confirm what the statistical analyses label as “significant” genes,
e.g. to ensure that the result is not biased by outliers (high counts) at individual sites,
but rather that apparent differences in counts between conditions are reflected as a consistent trend
over multiple TA sites in a gene.  <em>This is a recommended practice.</em></p>
<p>Once a combined_wig file is prepared, it can be used to
<a class="reference internal" href="transit_features.html#transit-quality-control"><span class="std std-ref">assess data quality</span></a>. There are two methods
available.  First, there is a <a class="reference internal" href="method_tnseq_stats.html#tnseq-stats"><span class="std std-ref">tnseq_stats</span></a> command, which will
calculate various metrics for each sample, include saturation
(density, percent of TA sites with non-zero insertions), mean read
count (NZmean), as well as skewness and other statistics of the
read-count distribution (for diagnostic purposes).  This can be run at
the command-line and in the GUI. Also, plots of read-count
distributions can be generated for selected samples in the GUI (again,
helpful for identifying highly skewed samples).  A discussion about
skewed samples, the problems they cause, and what to do about them can be
found <a class="reference internal" href="transit_features.html#transit-quality-control"><span class="std std-ref">here</span></a> (Quality Control).</p>
<p>One can also evaluate and compare samples by making <a class="reference internal" href="method_scatterplot.html#scatterplot"><span class="std std-ref">scatter plots</span></a> and pairwise <a class="reference internal" href="method_corrplot.html#corrplot"><span class="std std-ref">correlation plots</span></a>,
to get a preview of how samples are related to each other.</p>
<a class="reference internal image-reference" href="_images/glyc_chol_corrplot.png"><img alt="_images/glyc_chol_corrplot.png" class="align-center" src="_images/glyc_chol_corrplot.png" style="width: 300px;" /></a>
</div>
<div class="section" id="genome-annotations-prot-tables-and-gff-files">
<h2>Genome Annotations (.prot_tables and .gff files)<a class="headerlink" href="#genome-annotations-prot-tables-and-gff-files" title="Permalink to this headline">¶</a></h2>
<p>The annotation of a genome contains information about genes, such as
coordinates, strand, locus tag, gene name, and functional description.
Transit uses a custom format for annotations called prot_tables,
e.g. H37Rv.prot_table.  <a class="reference internal" href="input_files.html#annotation-files"><span class="std std-ref">Prot_tables</span></a> are <strong>tab-separated text files</strong>
containing the information on each gene, such as coordinates, strand, ORF id, and gene name.</p>
<p>In many cases, users might often obtain annotations for their genome
in .gff (or .gff3) file format, such as downloaded from NCBI.  .gff
files contains essentially the same information about genes.  However,
there is a bit more flexibility in the .gff file format (especially in
the tags used in the right-most column), and the information about
genes is not always encoded in a uniform way, making it difficult to
use arbitrary .gffs for analyses in Transit.  Therefore, there is a
simple procedure in Transit to convert a .gff file to .prot_table
format (see
<a class="reference internal" href="input_files.html#annotation-files"><span class="std std-ref">instructions for converting .gff files to .prot_tables</span></a>).
This step only has to be done once, and then the .prot_table can be used
for all subsequent analyses in Transit.</p>
</div>
<div class="section" id="gui">
<h2>GUI<a class="headerlink" href="#gui" title="Permalink to this headline">¶</a></h2>
<p>Here is a screenshot of the new GUI in Transit 4.0:</p>
<a class="reference internal image-reference" href="_images/Transit4.0_GUI.png"><img alt="_images/Transit4.0_GUI.png" class="align-center" src="_images/Transit4.0_GUI.png" style="width: 1000px;" /></a>
<p><strong>(note replace image when we update the version number; show it with an ‘resampling’ parameters and an output file in Results panel)</strong></p>
<p>Basic walk-through:</p>
<ul class="simple">
<li><p>You start by <strong>loading 3 input files</strong> (in succession): combined_wig, metadata, and annotation. This will populate the upper panel with individual samples, and the middle panel with conditions.</p></li>
<li><p>Next, you can evaluate certain samples by selecting them and then choose an Action from the ‘Select Tool’ dropdown box above the samples panel, such as displaying track views, making scatter plots, examining chromosomal bias via LOESS plots, and showing plots of read-count distributions (for quality control).</p></li>
<li><p>Next, you can select an analysis method from the menu (under <strong>Methods-&gt;himar1</strong>), such as gumbel, HMM, resampling, ZINB…  This will bring up a corresponding <strong>parameter panel</strong> on the right.  You might need to select specific samples or conditions to analyze. You can usually use the defaults for the other parameters.</p></li>
<li><p>Then you hit the <strong>Run button</strong>. You should be able to monitor progress via the Progress Bar.</p></li>
<li><p>Status updates and various messages will be displayed in the <strong>Message Bar</strong> at the bottom of the GUI window. Important: The full list of messages, including error messages will be printed on in the console window from where you started Transit. Check these message if anything goes wrong.</p></li>
<li><p>Usually, an analysis method will generate one or more output files.  These are typically text files in tab-separated format (which could be opened as spreadsheets in Excel). The file will get populated into the <strong>Results Table</strong> (bottom pannel).</p></li>
<li><p>If you select an output file in the panel, it will provide a dropdown with Actions you can perform, including  displaying the file as a table (or figure, if it is an image, such as volcano plots, heatmaps, etc). Some output files have customized Actions, such as making a volcano plot from output of resampling, or making a heatmap from the output file after running ANOVA or ZINB.</p></li>
<li><p>One of the most common and important Actions (‘Select Tool’ dropdown under Results Files) is to perform is <strong>Pathway Enrichment Analysis</strong> on the genes found to be significant by one of the other analyses (e.g. gumbel, hmm, resampling, ANOVA). Most of these output files have a column with adjusted P-values, and significant “hits” are usually defined as genes with Padj&lt;0.05.  If you have more than about 20 hits, you can use Pathway Enrichment Analysis to determine whether they share functional similarities.  There are several systems of pathways available, including COG categories and GO terms.</p></li>
</ul>
</div>
<div class="section" id="pre-processing-tasks">
<h2>Pre-Processing Tasks<a class="headerlink" href="#pre-processing-tasks" title="Permalink to this headline">¶</a></h2>
<p>When you first start the Transit GUI, you load up your TnSeq dataset (combined_wig file).
Before running any statistical analyses, the first thing you will probably want to
do is explore the individual samples, their relationships, and data quality.
Most of these steps can be performed by clicking on a sample in the samples menu
and selecting an action from the drop-down list, or by choosing one of the items from
the Pre-Processing menu.  These steps can also be done at the command line.</p>
<blockquote>
<div><ul class="simple">
<li><p>generate a <a class="reference internal" href="method_tnseq_stats.html#tnseq-stats"><span class="std std-ref">tnseq_stats</span></a> table (under Pre-Processing menu) summaring key statistics and metrics for each individual sample (including saturation, skewness, etc)</p></li>
<li><p>generate a <strong>LOESS plot</strong> (drop-down list) to see whether the mean read count variest significantly across the genome (chromosomal position bias; M. tuberculosis samples typically do not show a substantial bias)</p></li>
<li><p>examine <strong>distributions of read-counts and QQ-plots</strong> (select ‘Quality Control’ in drop-down list) to check for highly skewed samples</p></li>
<li><p>look at a <strong>Track View</strong> (drop-down list) for one or more samples that shows insertions (vertical bars) at TA sites in a target gene or locus</p></li>
<li><p>compute a <a class="reference internal" href="method_gene_means.html#gene-means"><span class="std std-ref">gene means</span></a> spreadsheet, with the mean insertion count in each gene in each condition (Pre-Processing-&gt;Gene Means) - this generates a helpful spreadsheet from which one can make barcharts showing how the (normalized) insertion counts vary across conditions for genes of interest</p></li>
<li><p><a class="reference internal" href="transit_normalization_tutorial.html#normalization"><span class="std std-ref">normalize</span></a> counts in a combined_wig file (Pre-Processing-&gt;Normalize) - while most of the analysis methods automatically perform TTR normalization, and even the method for creating combined_wig files normalizes by default, one can choose to normalize a sample or whole dataset a different way (such as the Beta-Geometric Correction, BGC).  But in most cases, users will not need to do an explicit normalization step.</p></li>
<li><p>make a <a class="reference internal" href="method_scatterplot.html#scatterplot"><span class="std std-ref">scatter plot</span></a> (Pre-Processing-&gt;Visuals-&gt;scatterplot) between 2 samples showing how correlated the mean insertion counts are at the gene level to check for consistency/reproducibility</p></li>
<li><p>make a <a class="reference internal" href="method_corrplot.html#corrplot"><span class="std std-ref">correlation plot</span></a> (Pre-Processing-&gt;Visuals-&gt;corrplot) among all samples to see which conditions appear more similar to each other, and to check that replicates are most highly correlated with each other (or check for outlier samples that do not correlate well with other replicates of the same condition, which might suggest they are bad or noisy)</p></li>
</ul>
</div></blockquote>
</div>
<div class="section" id="statistical-analysis-methods">
<h2>Statistical Analysis Methods<a class="headerlink" href="#statistical-analysis-methods" title="Permalink to this headline">¶</a></h2>
<p>The analysis methods available in Transit are divided into 3 categories:</p>
<ul class="simple">
<li><p>Methods for analyzing <strong>single conditions</strong> or datasets and identifying <em>essential genes</em></p>
<ul>
<li><p><a class="reference internal" href="method_gumbel.html#gumbel"><span class="std std-ref">Gumbel method</span></a> - looks for genes with larges ‘gaps’, or consecutive sequences of TA sites without insertions</p></li>
<li><p><a class="reference internal" href="method_HMM.html#hmm"><span class="std std-ref">Hidden Markov Model</span></a> (HMM) - assigns genes to one of 4 states: ES (essential), GD (growth-defect), NE (non-essential), or GA (growth-advantaged), based on magnitudes of insertion counts (reflecting fitness effects of mutants)</p></li>
</ul>
</li>
<li><p>Methods for <strong>pairwise comparisons of datasets</strong> (e.g. between a treatment and control condition) to identify <em>conditionally essential genes</em></p>
<ul>
<li><p><a class="reference internal" href="method_resampling.html#resampling"><span class="std std-ref">Resampling</span></a> - a non-parametric test using permutations of insertion counts to simulate a null distribution of difference in mean counts for each gene</p></li>
<li><p><a class="reference internal" href="method_Utest.html#utest"><span class="std std-ref">Mann-Whiney U-test</span></a> - another non-parametric test based on comparison of rank-ordering of insertion counts in each gene</p></li>
</ul>
</li>
<li><p>Methods for analyzing <strong>multiple conditions</strong> (&gt;=2) to identify genes which exhibit <em>significant variability</em> of insertion counts across conditions</p>
<ul>
<li><p><a class="reference internal" href="method_anova.html#anova"><span class="std std-ref">ANOVA</span></a></p></li>
<li><p><a class="reference internal" href="method_zinb.html#zinb"><span class="std std-ref">ZINB</span></a> - similar to ANOVA, but uses the Zero-Inflated Negative Binomial disribution to model counts; this method has options for testing for <em>interactions among covariates</em></p></li>
<li><p><a class="reference internal" href="method_GI.html#genetic-interactions"><span class="std std-ref">Genetic Interaction</span></a> analysis - this is a special case customized for testing interactions between 2 variables in a 2x2 experimental design (4 conditions)</p></li>
</ul>
</li>
</ul>
</div>
<div class="section" id="analyses-for-tn5-datasets">
<h2>Analyses for Tn5 Datasets<a class="headerlink" href="#analyses-for-tn5-datasets" title="Permalink to this headline">¶</a></h2>
<p>Although Transit was originally designed for analyzing Himar1 TnSeq datasets,
many of the methods can be adapted for analyzing datasets that use other transposons like Tn5.
The main difference is that the Himar1 transposon is restricted to insertions at
TA dinucleotide sites, whereas Tn5 is capable of inserting more broadly at any coordinate
in the genome.</p>
<p><strong>1/3/2023:</strong> Previously (in Transit versions up through 3.2.7), we
had a few statistical methods that were customized for analyzing Tn5
datasets.  These have been temporarily disabled during the transition
to Transit 4.0.  We will be adding back in the Tn5 analysis methods
soon… (in a future release, like 4.1)</p>
</div>
<div class="section" id="results-and-post-processing">
<h2>Results and Post-Processing<a class="headerlink" href="#results-and-post-processing" title="Permalink to this headline">¶</a></h2>
<p>Most of the analysis methods in Transit generate output files
that can be opened as spreadsheets in a program like Excel.
The output files are generally <strong>tab-separated</strong> text files, with
header lines demarcated by ‘#’ as prefixes.</p>
<p>For most of the analysis methods, the output file contains
a row for each gene in the genome with information relevant to
the statistical test, usually ending in columns labeled “P-value” and “Adj P-value” (Padj).
The P-value is calculated based on the statistical test, and then
all the P-values are adjusted by the Benjamini-Hochberg procedure
to correct for multiple testing (aiming to limit the false discovery rate to FDR&lt;5%).</p>
<p>In Transit, <strong>hits</strong> (or significant genes in the test) are generally
defined as those with <strong>Padj&lt;0.05</strong>, which can be identified in the
output files by sorting on the “Adj P-value” column.</p>
<p>After the user runs a TRANSIT Analysis Method, various functions can
be performed on the output file to better understand the results of
the analyis performed.  If using the GUI, the output file is visible
in the <em>Results Panel</em>, along with a summary (params and outcomes).
Click on the file and
select one of the following (availablity depends on analysis that was
run) from the action drop-down:</p>
<ul class="simple">
<li><p>Display Table - an external window will appear in an spreadsheet-like format for you to view the file</p></li>
<li><p>Volcano Plot - an external window will appear that displays a plot of the LFCs vs. log10(pvalue) with a horizontal line indicating the thresold of significance</p></li>
<li><p>Display Heatmap - an external window will appear of clustered conditions and significant hits resulting from the analysis. This file will be also saved to your folder of choice
and placed in the results pane, which then can be viewed by selection of the “View” option in the action dropdown.</p></li>
<li><p><a class="reference internal" href="method_pathway_enrichment.html#gsea"><span class="std std-ref">Pathway Enrichment</span></a> Analysis - this will search for significantly enriched pathways among the hits (Padj&lt;0.05) in the selected file in the Results Panel (e.g. an output file from an analysis like resampling, ANOVA, etc)</p></li>
</ul>
</div>
<div class="section" id="command-line">
<h2>Command Line<a class="headerlink" href="#command-line" title="Permalink to this headline">¶</a></h2>
<p>The analysis methods in Transit are also described in this <a class="reference external" href="https://orca1.tamu.edu/essentiality/transit/transit-manual.pdf">PDF manual</a> , focusing on
command-line operations.</p>
<p>Most of the methods in Transit can be run from the command line.
Typically, you run this as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python</span> <span class="n">TRANSIT_PATH</span><span class="o">/</span><span class="n">src</span><span class="o">/</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="o">&lt;</span><span class="n">command</span><span class="o">&gt;</span> <span class="n">args</span><span class="o">...</span>
</pre></div>
</div>
<p>Remember to use python3.</p>
<p>Commands include: gumbel, resampling, hmm, GI, tnseq_stats, anova…
If you run ‘python TRANSIT_PATH/src/transit.py –help’, it will print out the full list of available commands.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python</span> <span class="n">TRANSIT_PATH</span><span class="o">/</span><span class="n">src</span><span class="o">/</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="o">--</span><span class="n">help</span>
<span class="n">The</span> <span class="n">available</span> <span class="n">subcommands</span> <span class="n">are</span><span class="p">:</span>
   <span class="n">anova</span>
   <span class="n">gi</span>
   <span class="n">gumbel</span>
   <span class="n">hmm</span>
   <span class="o">...</span>
</pre></div>
</div>
<p>The arguments would be whatever input files, options, and flags are appropriate for a given command.</p>
<p>If you want a reminder of the <strong>usage</strong> for given command, use run that command without any arguments.
For example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python</span> <span class="n">TRANSIT_PATH</span><span class="o">/</span><span class="n">src</span><span class="o">/</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">anova</span>
<span class="n">Usage</span><span class="p">:</span>
   <span class="n">python</span> <span class="n">src</span><span class="o">/</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">anova</span> <span class="o">&lt;</span><span class="n">combined_wig_file</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">annotation_file</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">metadata_file</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">output_file</span><span class="o">&gt;</span> <span class="p">[</span><span class="n">Optional</span> <span class="n">Arguments</span><span class="p">]</span>

<span class="n">Optional</span> <span class="n">Arguments</span><span class="p">:</span>
  <span class="o">-</span><span class="n">include</span><span class="o">-</span><span class="n">conditions</span> <span class="o">&lt;</span><span class="n">cond1</span><span class="p">,</span><span class="o">...&gt;</span> <span class="o">:=</span> <span class="n">Comma</span><span class="o">-</span><span class="n">separated</span> <span class="nb">list</span> <span class="n">of</span> <span class="n">conditions</span> <span class="n">to</span> <span class="n">use</span> <span class="k">for</span> <span class="n">analysis</span> <span class="p">(</span><span class="n">Default</span><span class="p">:</span> <span class="nb">all</span><span class="p">)</span>
  <span class="o">-</span><span class="n">exclude</span><span class="o">-</span><span class="n">conditions</span> <span class="o">&lt;</span><span class="n">cond1</span><span class="p">,</span><span class="o">...&gt;</span> <span class="o">:=</span> <span class="n">Comma</span><span class="o">-</span><span class="n">separated</span> <span class="nb">list</span> <span class="n">of</span> <span class="n">conditions</span> <span class="n">to</span> <span class="n">exclude</span> <span class="p">(</span><span class="n">Default</span><span class="p">:</span> <span class="n">none</span><span class="p">)</span>
  <span class="o">-</span><span class="n">n</span> <span class="o">&lt;</span><span class="n">string</span><span class="o">&gt;</span> <span class="o">:=</span> <span class="n">Normalization</span> <span class="n">method</span><span class="o">.</span> <span class="n">Default</span><span class="p">:</span> <span class="o">-</span><span class="n">n</span> <span class="n">TTR</span>
  <span class="o">-</span><span class="n">ref</span> <span class="o">&lt;</span><span class="n">cond</span><span class="o">&gt;</span> <span class="o">:=</span> <span class="n">which</span> <span class="n">condition</span><span class="p">(</span><span class="n">s</span><span class="p">)</span> <span class="n">to</span> <span class="n">use</span> <span class="k">as</span> <span class="n">a</span> <span class="n">reference</span> <span class="k">for</span> <span class="n">calculating</span> <span class="n">LFCs</span> <span class="p">(</span><span class="n">comma</span><span class="o">-</span><span class="n">separated</span> <span class="k">if</span> <span class="n">multiple</span> <span class="n">conditions</span><span class="p">)</span>
  <span class="o">-</span><span class="n">iN</span>    <span class="o">&lt;</span><span class="n">N</span><span class="o">&gt;</span>  <span class="o">:=</span> <span class="n">Ignore</span> <span class="n">TAs</span> <span class="n">within</span> <span class="n">given</span> <span class="n">percentage</span> <span class="p">(</span><span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span> <span class="mi">5</span><span class="p">)</span> <span class="n">of</span> <span class="n">N</span> <span class="n">terminus</span><span class="o">.</span> <span class="n">Default</span><span class="p">:</span> <span class="o">-</span><span class="n">iN</span> <span class="mi">0</span>
  <span class="o">-</span><span class="n">iC</span>    <span class="o">&lt;</span><span class="n">N</span><span class="o">&gt;</span>  <span class="o">:=</span> <span class="n">Ignore</span> <span class="n">TAs</span> <span class="n">within</span> <span class="n">given</span> <span class="n">percentage</span> <span class="p">(</span><span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span> <span class="mi">5</span><span class="p">)</span> <span class="n">of</span> <span class="n">C</span> <span class="n">terminus</span><span class="o">.</span> <span class="n">Default</span><span class="p">:</span> <span class="o">-</span><span class="n">iC</span> <span class="mi">0</span>
  <span class="o">-</span><span class="n">PC</span>    <span class="o">&lt;</span><span class="n">N</span><span class="o">&gt;</span>  <span class="o">:=</span> <span class="n">pseudocounts</span> <span class="n">to</span> <span class="n">use</span> <span class="k">for</span> <span class="n">calculating</span> <span class="n">LFCs</span><span class="o">.</span> <span class="n">Default</span><span class="p">:</span> <span class="o">-</span><span class="n">PC</span> <span class="mi">5</span>
  <span class="o">-</span><span class="n">alpha</span> <span class="o">&lt;</span><span class="n">N</span><span class="o">&gt;</span>  <span class="o">:=</span> <span class="n">value</span> <span class="n">added</span> <span class="n">to</span> <span class="n">mse</span> <span class="ow">in</span> <span class="n">F</span><span class="o">-</span><span class="n">test</span> <span class="k">for</span> <span class="n">moderated</span> <span class="n">anova</span> <span class="p">(</span><span class="n">makes</span> <span class="n">genes</span> <span class="k">with</span> <span class="n">low</span> <span class="n">counts</span> <span class="n">less</span> <span class="n">significant</span><span class="p">)</span><span class="o">.</span> <span class="n">Default</span><span class="p">:</span> <span class="o">-</span><span class="n">alpha</span> <span class="mi">1000</span>
  <span class="o">--</span><span class="n">winz</span>      <span class="o">:=</span> <span class="n">winsorize</span> <span class="n">insertion</span> <span class="n">counts</span> <span class="k">for</span> <span class="n">each</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">each</span> <span class="n">condition</span> <span class="p">(</span><span class="n">replace</span> <span class="nb">max</span> <span class="n">cnt</span> <span class="k">with</span> <span class="mi">2</span><span class="n">nd</span> <span class="n">highest</span><span class="p">;</span> <span class="n">helps</span> <span class="n">mitigate</span> <span class="n">effect</span> <span class="n">of</span> <span class="n">outliers</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="developers">
<h2>Developers<a class="headerlink" href="#developers" title="Permalink to this headline">¶</a></h2>
<table class="docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 11%" />
<col style="width: 69%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Name</p></th>
<th class="head"><p>Time Active</p></th>
<th class="head"><p>Contact Information</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Thomas R. Ioerger</p></td>
<td><p>2015-Present</p></td>
<td><p><a class="reference external" href="http://faculty.cs.tamu.edu/ioerger/">http://faculty.cs.tamu.edu/ioerger/</a></p></td>
</tr>
<tr class="row-odd"><td><p>Michael A. DeJesus</p></td>
<td><p>2015-2018</p></td>
<td><p><a class="reference external" href="http://mad-lab.org">http://mad-lab.org</a></p></td>
</tr>
<tr class="row-even"><td><p>Chaitra Ambadipudi</p></td>
<td><p>2015</p></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Eric Nelson</p></td>
<td><p>2016</p></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Siddharth Subramaniyam</p></td>
<td><p>2018</p></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Sanjeevani Choudhery</p></td>
<td><p>2021-</p></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Jeff Hykin</p></td>
<td><p>2022-</p></td>
<td></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<p>If you use TRANSIT, please cite the following reference:</p>
<dl class="citation">
<dt class="label" id="dejesus2015transit"><span class="brackets">DeJesus2015TRANSIT</span></dt>
<dd><p><a class="reference external" href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004401">DeJesus, M.A., Ambadipudi, C., Baker, R., Sassetti, C., and Ioerger, T.R. (2015). TRANSIT - a Software Tool for Himar1 TnSeq Analysis. PLOS Computational Biology, 11(10):e1004401</a></p>
</dd>
</dl>
<p>Development of TRANSIT is funded by the National Institutes of Health (www.nih.gov/) grant U19 AI107774.</p>
<p>Other references for methods utilized by TRANSIT:</p>
<dl class="citation">
<dt class="label" id="dejesus2013"><span class="brackets">DeJesus2013</span></dt>
<dd><p><a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/23361328">DeJesus, M.A., Zhang, Y.J., Sassettti, C.M., Rubin, E.J.,
Sacchettini, J.C., and Ioerger, T.R. (2013). Bayesian analysis of gene essentiality based on sequencing of transposon insertion libraries. Bioinformatics, 29(6):695-703.</a></p>
</dd>
<dt class="label" id="dejesus2013hmm"><span class="brackets">DeJesus2013HMM</span></dt>
<dd><p><a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/24103077">DeJesus, M.A., Ioerger, T.R. A Hidden Markov Model for identifying essential and growth-defect regions in bacterial genomes from transposon insertion sequencing data. BMC Bioinformatics. 2013. 14:303</a></p>
</dd>
<dt class="label" id="dejesus2014"><span class="brackets">DeJesus2014</span></dt>
<dd><p><a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/26357081">DeJesus, M.A. and Ioerger, T.R. (2014). Capturing uncertainty by modeling local transposon insertion frequencies improves discrimination of essential genes. IEEE Transactions on Computational Biology and Bioinformatics, 12(1):92-102.</a></p>
</dd>
<dt class="label" id="dejesus2016"><span class="brackets">DeJesus2016</span></dt>
<dd><p><a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/26932272">DeJesus, M.A. and Ioerger, T.R. (2016). Normalization of transposon-mutant library sequencing datasets to improve identification of conditionally essential genes. Journal of Bioinformatics and Computational Biology, 14(3):1642004</a></p>
</dd>
<dt class="label" id="dejesus2017nar"><span class="brackets">DeJesus2017NAR</span></dt>
<dd><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/28334803">DeJesus, M.A., Nambi, S., Smith, C.M., Baker, R.E., Sassetti, C.M., Ioerger, T.R. Statistical analysis of genetic interactions in Tn-Seq data.  Nucleic Acids Research. 2017. 45(11):e93. doi: 10.1093/nar/gkx128.</a></p>
</dd>
<dt class="label" id="zinb"><span class="brackets">ZINB</span></dt>
<dd><p><a class="reference external" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-3156-z">Subramaniyam S, DeJesus MA, Zaveri A, Smith CM, Baker RE, Ehrt S, Schnappinger D, Sassetti CM, Ioerger TR. (2019).  Statistical analysis of variability in TnSeq data across conditions using Zero-Inflated Negative Binomial regression. *BMC Bioinformatics*. 2019 Nov 21;20(1):603. doi: 10.1186/s12859-019-3156-z.</a></p>
</dd>
<dt class="label" id="choudhery2021"><span class="brackets">Choudhery2021</span></dt>
<dd><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/34665010/">Choudhery S, Brown AJ, Akusobi C, Rubin EJ, Sassetti CM, Ioerger TR. Modeling Site-Specific Nucleotide Biases Affecting Himar1 Transposon Insertion Frequencies in TnSeq Data Sets. *mSystems*. 2021 Oct 26;6(5):e0087621. doi: 10.1128/mSystems.00876-21.</a></p>
</dd>
</dl>
</div>
</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="transit_install.html" class="btn btn-neutral float-right" title="Installation" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
        <a href="transit_overview.html" class="btn btn-neutral float-left" title="TRANSIT Overview" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2015, Michael A. DeJesus &amp; Thomas R. Ioerger.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(false);
      });
  </script>

  
  
    
   

</body>
</html>