

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>tnseq_stats &mdash; TRANSIT v3.3.1 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/default.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> TRANSIT
          

          
            
            <img src="_static/transit_logo.png" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                v3.3.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption" role="heading"><span class="caption-text">TRANSIT MANUAL</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="transit_overview.html">TRANSIT Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_install.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_running.html">Running TRANSIT</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_features.html">Quality Control</a></li>
<li class="toctree-l1"><a class="reference internal" href="file_formats.html">File Formats for Transit</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">[** NEW **] CRISRPi ANALYSIS METHODS</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="CGI.html">CRISPRi-DR</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">TnSeq ANALYSIS METHODS</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="method_gumbel.html">Gumbel</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_griffin.html">Griffin</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_tn5gaps.html">Tn5Gaps</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_HMM.html">HMM</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_resampling.html">Resampling</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_Utest.html">Mann-Whitney U-test (utest)</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_GI.html">Genetic Interactions</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_anova.html">ANOVA</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_zinb.html">ZINB</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_normalization.html">Normalization</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_pathway_enrichment.html">Pathway Enrichment Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_corrplot.html">corrplot</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_heatmap.html">heatmap</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_ttnfitness.html">TTN-Fitness</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">TRANSIT Tutorials</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="transit_interactions_tutorial.html">Tutorial: Genetic Interactions Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_normalization_tutorial.html">Tutorial: Normalize datasets</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_export_tutorial.html">Tutorial: Export datasets</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_console_cheatsheet.html">Console Mode Cheat-Sheet</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">TPP Manual</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="tpp.html">TPP Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#installation">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#running-tpp">Running TPP</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#mapping-to-genomes-with-multiple-contigs">Mapping to Genomes with Multiple Contigs</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#overview-of-data-processing-procedure">Overview of Data Processing Procedure</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#statistics">Statistics</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Code Documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="transit.html">pytransit package</a></li>
<li class="toctree-l1"><a class="reference internal" href="example_scripts.html">Example Scripts (for Developers)</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">TRANSIT</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>tnseq_stats</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/methods_tnseq_stats.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="tnseq-stats">
<span id="id1"></span><h1>tnseq_stats<a class="headerlink" href="#tnseq-stats" title="Permalink to this headline">¶</a></h1>
<p>This command is useful for calculating some metrics on input wig (or
combined_wig files) for assessement of data quality.</p>
<div class="section" id="usage">
<h2>Usage:<a class="headerlink" href="#usage" title="Permalink to this headline">¶</a></h2>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">usage</span><span class="p">:</span> <span class="n">python3</span> <span class="n">src</span><span class="o">/</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">tnseq_stats</span> <span class="o">&lt;</span><span class="n">file</span><span class="o">.</span><span class="n">wig</span><span class="o">&gt;+</span> <span class="p">[</span><span class="o">-</span><span class="n">o</span> <span class="o">&lt;</span><span class="n">output_file</span><span class="o">&gt;</span><span class="p">]</span>
         <span class="n">python</span> <span class="n">src</span><span class="o">/</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">tnseq_stats</span> <span class="o">-</span><span class="n">c</span> <span class="o">&lt;</span><span class="n">combined_wig</span><span class="o">&gt;</span> <span class="p">[</span><span class="o">-</span><span class="n">o</span> <span class="o">&lt;</span><span class="n">output_file</span><span class="o">&gt;</span><span class="p">]</span>
</pre></div>
</div>
<p>It generates a table (tab-separated text file that can be opened in Excel) with the following statistics in it:</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 7%" />
<col style="width: 25%" />
<col style="width: 67%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Column Header</p></th>
<th class="head"><p>Column Definition</p></th>
<th class="head"><p>Comments</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Filename</p></td>
<td><p>Name of sample (wig file)</p></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Density</p></td>
<td><p>Fraction of sites with insertions.</p></td>
<td><p>“Well-saturated” Himar1 datasets have &gt;30% saturation. Below this, statistical methods may have trouble.</p></td>
</tr>
<tr class="row-even"><td><p>mean_ct</p></td>
<td><p>Average read-count over all TA sites.</p></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>NZMean</p></td>
<td><p>Average read-count, excluding empty sites.</p></td>
<td><p>A value between 30-200 is usually good for Himar1 datasets. Too high or too low can indicate problems.</p></td>
</tr>
<tr class="row-even"><td><p>NZMedian</p></td>
<td><p>Median read-count, excluding empty sites.</p></td>
<td><p>As read-counts can often have spikes, median serves as a good robust estimate.</p></td>
</tr>
<tr class="row-odd"><td><p>max_ct</p></td>
<td><p>Largest read-count at any TA site</p></td>
<td><p>Useful to determine whether there are outliers/spikes, which may indicate sequencing issues.</p></td>
</tr>
<tr class="row-even"><td><p>total_cts</p></td>
<td><p>Sum of total read-counts in the sample.</p></td>
<td><p>Indicates how much sequencing material was obtained. Typically &gt;1M reads is desired for Himar1 datasets.</p></td>
</tr>
<tr class="row-odd"><td><p>skewness</p></td>
<td><p>3rd-order moment of read-count distribution.</p></td>
<td><p>Sharp peak? Large skew may indicate issues with a dataset. Typically a skew &lt; 50 is desired. May be higher when</p></td>
</tr>
<tr class="row-even"><td><p>kurtosis</p></td>
<td><p>4th-order moment of read-counts distribution</p></td>
<td><p>Lop-sided peak?</p></td>
</tr>
<tr class="row-odd"><td><p>PTI</p></td>
<td><p>Pickand Tail Index</p></td>
<td><p>Another statistical measure that indicates presence of individual TA sites with high outlier counts. PTI&gt;1.0 is not good.</p></td>
</tr>
</tbody>
</table>
<p>Here is an example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python3</span> <span class="n">src</span><span class="o">/</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">tnseq_stats</span> <span class="o">-</span><span class="n">c</span> <span class="n">src</span><span class="o">/</span><span class="n">pytransit</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">cholesterol_glycerol_combined</span><span class="o">.</span><span class="n">dat</span>
<span class="n">dataset</span> <span class="n">density</span> <span class="n">mean_ct</span> <span class="n">NZmean</span>  <span class="n">NZmedian</span>        <span class="n">max_ct</span>  <span class="n">total_cts</span>       <span class="n">skewness</span>        <span class="n">kurtosis</span>        <span class="n">pickands_tail_index</span>
<span class="n">src</span><span class="o">/</span><span class="n">pytransit</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">cholesterol_H37Rv_rep1</span><span class="o">.</span><span class="n">wig</span>   <span class="mf">0.439</span>   <span class="mf">139.6</span>   <span class="mf">317.6</span>   <span class="mi">147</span>     <span class="mf">125355.5</span>        <span class="mi">10414005</span>        <span class="mf">54.8</span>    <span class="mf">4237.7</span>  <span class="mf">0.973</span>
<span class="n">src</span><span class="o">/</span><span class="n">pytransit</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">cholesterol_H37Rv_rep2</span><span class="o">.</span><span class="n">wig</span>   <span class="mf">0.439</span>   <span class="mf">171.4</span>   <span class="mf">390.5</span>   <span class="mi">148</span>     <span class="mf">704662.8</span>        <span class="mi">12786637</span>        <span class="mf">105.8</span>   <span class="mf">14216.2</span> <span class="mf">1.529</span>
<span class="n">src</span><span class="o">/</span><span class="n">pytransit</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">cholesterol_H37Rv_rep3</span><span class="o">.</span><span class="n">wig</span>   <span class="mf">0.359</span>   <span class="mf">173.8</span>   <span class="mf">484.2</span>   <span class="mi">171</span>     <span class="mf">292294.8</span>        <span class="mi">12968502</span>        <span class="mf">42.2</span>    <span class="mf">2328.0</span>  <span class="mf">1.584</span>
<span class="n">src</span><span class="o">/</span><span class="n">pytransit</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">glycerol_H37Rv_rep1</span><span class="o">.</span><span class="n">wig</span>      <span class="mf">0.419</span>   <span class="mf">123.3</span>   <span class="mf">294.5</span>   <span class="mi">160</span>     <span class="mf">8813.3</span>  <span class="mi">9195672</span> <span class="mf">4.0</span>     <span class="mf">33.0</span>    <span class="mf">0.184</span>
<span class="n">src</span><span class="o">/</span><span class="n">pytransit</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">glycerol_H37Rv_rep2</span><span class="o">.</span><span class="n">wig</span>      <span class="mf">0.516</span>   <span class="mf">123.8</span>   <span class="mf">240.1</span>   <span class="mi">127</span>     <span class="mf">8542.5</span>  <span class="mi">9235984</span> <span class="mf">4.0</span>     <span class="mf">33.5</span>    <span class="mf">0.152</span>
</pre></div>
</div>
<p>In this example, you can see the 5 samples have saturations in the range of 35.9-51.6% (which is descent).
The NZMeans are 123-139, but this is <em>post-normalization</em>.
(TTR normalization had already been applied to this combined_wig file, so the means can be expected to be scaled to around 100.)
If you want to see the NZmeans for the <em>raw data</em>, re-generate the combined_wig file using ‘-n nonorm’, to skip the automatic normalization step.
These samples also exhibit <em>skewness</em> that is on the high side (33-105.8).
This is probably related to the fact that some individual TA sites have very high counts.
For example, rep2 of cholesterol has a max count of 704662 at a single TA site, representing over 5% of the 12.78M reads (total insertion counts).
TTR is supposed to be robust by ignoring the top 5% of most abundant sites, but still the rest of the distribution of counts could be skewed.
This sample also has a high <em>Pickands’ tail index</em> (1.53, which is above 1.0).
While, we currently don’t have recommendations for hard cutoffs to use for identifying bad samples (that might need to be re-sequenced),
I would say that skew&gt;30 and/or PTI&gt;1.0 suggest that samples are noisy or lower-quality.
See <cite>Quality Control &lt;transit_quality_control&gt;</cite> for more discussion of assessing quality of TnSeq datasets.
However, doing resampling on this data still yielded insights into many genes required for cholesterol metabolism in <em>M. tuberculosis</em>. [REF]</p>
</div>
</div>


           </div>
           
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2015, Michael A. DeJesus &amp; Thomas R. Ioerger.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>