

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>Genetic Interactions &mdash; TRANSIT v4.0.0 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/default.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="ANOVA" href="anova_method.html" />
    <link rel="prev" title="Mann-Whitney U-test (utest)" href="utest_method.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> TRANSIT
          

          
            
            <img src="_static/transit_logo.png" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                v4.0.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption" role="heading"><span class="caption-text">TRANSIT MANUAL</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="transit_overview.html">TRANSIT Overview (4.0)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="transit_overview.html#typical-workflow">Typical Workflow</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_overview.html#tpp-transit-pre-processor">TPP (Transit Pre-Processor)</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_overview.html#combined-wig-and-metadata-files">Combined_wig and Metadata Files</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_overview.html#genome-annotations-prot-tables-and-gff-files">Genome Annotations (.prot_tables and .gff files)</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_overview.html#gui">GUI</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_overview.html#pre-processing-tasks">Pre-Processing Tasks</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_overview.html#statistical-analysis-methods">Statistical Analysis Methods</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_overview.html#analyses-for-tn5-datasets">Analyses for Tn5 Datasets</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_overview.html#results-and-post-processing">Results and Post-Processing</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_overview.html#command-line">Command Line</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_overview.html#developers">Developers</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_overview.html#references">References</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="transit_install.html">Installation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="transit_install.html#requirements">Requirements</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_install.html#python-3">Python 3:</a><ul>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#additional-requirements-r-statistical-analysis-package">Additional Requirements: R (statistical analysis package)</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="transit_install.html#optional-install-bwa-to-use-with-tpp-pre-processor">Optional: Install BWA to use with TPP pre-processor</a><ul>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#linux-osx-instructions">Linux &amp; OSX Instructions</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#windows-instructions">Windows Instructions</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="transit_install.html#upgrading">Upgrading</a><ul>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#method-1-upgrading-package-installation">Method 1: Upgrading package installation</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#method-2-upgrading-source-installation">Method 2: Upgrading source installation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="transit_install.html#installing-wxpython">Installing wxPython</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_install.html#troubleshooting">Troubleshooting</a><ul>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#no-window-appears-when-running-in-gui-mode">1. No window appears when running in GUI mode.</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#pip3-systemerror-cannot-compile-python-h">2. pip3: SystemError: Cannot compile ‘Python.h’.</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#pip-the-following-required-packages-can-not-be-built-freetype-png-etc">3. pip: “The following required packages can not be built: freetype,png,” etc.</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#pip3-no-lapack-blas-resources-found">4. pip3: “No lapack/blas resources found”</a></li>
<li class="toctree-l3"><a class="reference internal" href="transit_install.html#resources-contextualversionconflict-six-1-5-2">5. “resources.ContextualVersionConflict (six 1.5.2)…”</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="transit_running.html">Running TRANSIT</a><ul>
<li class="toctree-l2"><a class="reference internal" href="transit_running.html#gui-mode">GUI Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit_running.html#command-line-mode">Command line Mode</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="input_files.html">Input Files for Transit</a><ul>
<li class="toctree-l2"><a class="reference internal" href="input_files.html#combined-wig-files">Combined_wig Files</a></li>
<li class="toctree-l2"><a class="reference internal" href="input_files.html#samples-metadata-file">Samples Metadata File</a></li>
<li class="toctree-l2"><a class="reference internal" href="input_files.html#genome-annotations-prot-tables-and-gff-files">Genome Annotations (.prot_tables and .gff files)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="quality_control.html">Quality Control</a><ul>
<li class="toctree-l2"><a class="reference internal" href="quality_control.html#qc-metrics-table">QC Metrics Table</a></li>
<li class="toctree-l2"><a class="reference internal" href="quality_control.html#qc-plots">QC Plots</a><ul>
<li class="toctree-l3"><a class="reference internal" href="quality_control.html#figure-1-read-count-distribution">Figure 1: Read-Count Distribution</a></li>
<li class="toctree-l3"><a class="reference internal" href="quality_control.html#figure-2-qq-plot-of-read-counts-vs-geometric-distribution">Figure 2: QQ-Plot of Read-Counts vs Geometric Distribution</a></li>
<li class="toctree-l3"><a class="reference internal" href="quality_control.html#figure-3-ranked-plot-of-read-counts">Figure 3: Ranked plot of Read-Counts</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="quality_control.html#interpretation-of-data-quality">Interpretation of Data Quality</a></li>
<li class="toctree-l2"><a class="reference internal" href="quality_control.html#what-to-do-about-skewed-samples">What to do about Skewed Samples?</a></li>
<li class="toctree-l2"><a class="reference internal" href="quality_control.html#beta-geometric-correction">Beta-Geometric Correction</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">PRE-PROCESSING</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="tnseq_stats_method.html">tnseq_stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="scatterplot_method.html">Scatterplot</a><ul>
<li class="toctree-l2"><a class="reference internal" href="scatterplot_method.html#command-line-usage">Command Line Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="scatterplot_method.html#example-usage">Example Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="scatterplot_method.html#gui-mode">GUI Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="scatterplot_method.html#runtime">Runtime</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="corrplot_method.html">Corrplot</a><ul>
<li class="toctree-l2"><a class="reference internal" href="corrplot_method.html#command-line-usage">Command Line Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="corrplot_method.html#gui-mode">GUI Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="corrplot_method.html#runtime">Runtime</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="normalize_method.html">Normalization</a><ul>
<li class="toctree-l2"><a class="reference internal" href="normalize_method.html#command-line">Command-line</a></li>
<li class="toctree-l2"><a class="reference internal" href="normalize_method.html#gui">GUI</a></li>
<li class="toctree-l2"><a class="reference internal" href="normalize_method.html#examples">Examples</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="gene_means_method.html">Gene Means</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">ANALYSES of single conditions</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="gumbel_method.html">Gumbel</a><ul>
<li class="toctree-l2"><a class="reference internal" href="gumbel_method.html#how-does-it-work">How does it work?</a></li>
<li class="toctree-l2"><a class="reference internal" href="gumbel_method.html#usage">Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="gumbel_method.html#parameters">Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="gumbel_method.html#gui-mode">GUI Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="gumbel_method.html#outputs-and-diagnostics">Outputs and diagnostics</a></li>
<li class="toctree-l2"><a class="reference internal" href="gumbel_method.html#run-time">Run-time</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="hmm_method.html">HMM</a><ul>
<li class="toctree-l2"><a class="reference internal" href="hmm_method.html#how-does-it-work">How does it work?</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmm_method.html#usage">Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmm_method.html#parameters">Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmm_method.html#gui-mode">GUI Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmm_method.html#output-and-diagnostics">Output and Diagnostics</a><ul>
<li class="toctree-l3"><a class="reference internal" href="hmm_method.html#sites-output-file">Sites Output File:</a></li>
<li class="toctree-l3"><a class="reference internal" href="hmm_method.html#genes-output-file">Genes Output File:</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="hmm_method.html#run-time">Run-time</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="ttnfitness_method.html">TTN-Fitness</a><ul>
<li class="toctree-l2"><a class="reference internal" href="ttnfitness_method.html#usage">Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="ttnfitness_method.html#gui-mode">GUI Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="ttnfitness_method.html#volcano-plot">Volcano Plot</a></li>
<li class="toctree-l2"><a class="reference internal" href="ttnfitness_method.html#output">Output</a><ul>
<li class="toctree-l3"><a class="reference internal" href="ttnfitness_method.html#genes-output-file">Genes Output File:</a></li>
<li class="toctree-l3"><a class="reference internal" href="ttnfitness_method.html#sites-output-file">Sites Output File:</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="ttnfitness_method.html#example-of-running-the-ttn-fitness-methodology-on-the-sample-glycerol-data">Example of running the TTN-Fitness methodology on the sample glycerol data</a></li>
<li class="toctree-l2"><a class="reference internal" href="ttnfitness_method.html#run-time">Run-time</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">ANALYSES for pairwise comparisons</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="resampling_method.html">Resampling</a><ul>
<li class="toctree-l2"><a class="reference internal" href="resampling_method.html#how-does-it-work">How does it work?</a></li>
<li class="toctree-l2"><a class="reference internal" href="resampling_method.html#site-restricted-s-r-resampling">Site-Restricted (S-R) Resampling</a></li>
<li class="toctree-l2"><a class="reference internal" href="resampling_method.html#usage">Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="resampling_method.html#parameters">Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="resampling_method.html#notes">Notes</a></li>
<li class="toctree-l2"><a class="reference internal" href="resampling_method.html#command-line-examples">Command Line Examples</a></li>
<li class="toctree-l2"><a class="reference internal" href="resampling_method.html#doing-resampling-with-datasets-from-different-libraries">Doing resampling with datasets from different libraries</a></li>
<li class="toctree-l2"><a class="reference internal" href="resampling_method.html#doing-resampling-between-different-strains">Doing resampling between different strains</a></li>
<li class="toctree-l2"><a class="reference internal" href="resampling_method.html#gui-mode">GUI Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="resampling_method.html#output-and-diagnostics">Output and Diagnostics</a></li>
<li class="toctree-l2"><a class="reference internal" href="resampling_method.html#run-time">Run-time</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="utest_method.html">Mann-Whitney U-test (utest)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="utest_method.html#usage">Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="utest_method.html#gui-mode">GUI Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="utest_method.html#output">Output</a></li>
<li class="toctree-l2"><a class="reference internal" href="utest_method.html#runtime">Runtime</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">ANALYSES for multiple conditions</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Genetic Interactions</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#how-does-it-work">How does it work?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#statistical-significance">Statistical Significance</a></li>
<li class="toctree-l2"><a class="reference internal" href="#usage">Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="#example">Example</a></li>
<li class="toctree-l2"><a class="reference internal" href="#parameters">Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="#gui-mode">Gui Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="#output-and-diagnostics">Output and Diagnostics</a></li>
<li class="toctree-l2"><a class="reference internal" href="#run-time">Run-time</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="anova_method.html">ANOVA</a><ul>
<li class="toctree-l2"><a class="reference internal" href="anova_method.html#how-does-it-work">How does it work?</a></li>
<li class="toctree-l2"><a class="reference internal" href="anova_method.html#example">Example</a></li>
<li class="toctree-l2"><a class="reference internal" href="anova_method.html#input-files">Input files</a></li>
<li class="toctree-l2"><a class="reference internal" href="anova_method.html#parameters">Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="anova_method.html#output-and-diagnostics">Output and Diagnostics</a></li>
<li class="toctree-l2"><a class="reference internal" href="anova_method.html#gui-mode">GUI Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="anova_method.html#run-time">Run-time</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="zinb_method.html">ZINB</a><ul>
<li class="toctree-l2"><a class="reference internal" href="zinb_method.html#how-does-it-work">How does it work?</a></li>
<li class="toctree-l2"><a class="reference internal" href="zinb_method.html#example">Example</a></li>
<li class="toctree-l2"><a class="reference internal" href="zinb_method.html#input-files">Input files</a></li>
<li class="toctree-l2"><a class="reference internal" href="zinb_method.html#parameters">Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="zinb_method.html#covariates-and-interactions">Covariates and Interactions</a></li>
<li class="toctree-l2"><a class="reference internal" href="zinb_method.html#categorical-vs-numeric-covariates">Categorical vs Numeric Covariates</a></li>
<li class="toctree-l2"><a class="reference internal" href="zinb_method.html#statistical-significance-what-the-p-values-mean-in-the-zinb-output">Statistical Significance - What the P-values Mean in the ZINB Output</a></li>
<li class="toctree-l2"><a class="reference internal" href="zinb_method.html#gui-mode">GUI Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="zinb_method.html#output-and-diagnostics">Output and Diagnostics</a></li>
<li class="toctree-l2"><a class="reference internal" href="zinb_method.html#run-time">Run-time</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">POST-PROCESSING</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="pathway_enrichment_method.html">Pathway Enrichment Analysis</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pathway_enrichment_method.html#command-line-usage">Command Line Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="pathway_enrichment_method.html#parameters">Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="pathway_enrichment_method.html#auxilliary-pathway-files-in-transit-data-directory">Auxilliary Pathway Files in Transit Data Directory</a></li>
<li class="toctree-l2"><a class="reference internal" href="pathway_enrichment_method.html#current-recommendations">Current Recommendations</a></li>
<li class="toctree-l2"><a class="reference internal" href="pathway_enrichment_method.html#examples">Examples</a></li>
<li class="toctree-l2"><a class="reference internal" href="pathway_enrichment_method.html#gui-mode">GUI Mode</a></li>
<li class="toctree-l2"><a class="reference internal" href="pathway_enrichment_method.html#output-and-diagnostics">Output and Diagnostics</a></li>
<li class="toctree-l2"><a class="reference internal" href="pathway_enrichment_method.html#run-time">Run-time</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="heatmap_method.html">Heatmap</a><ul>
<li class="toctree-l2"><a class="reference internal" href="heatmap_method.html#command-line-usage">Command Line Usage:</a></li>
<li class="toctree-l2"><a class="reference internal" href="heatmap_method.html#gui">GUI:</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">TPP Manual</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="tpp.html">TPP Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#installation">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#running-tpp">Running TPP</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#mapping-to-genomes-with-multiple-contigs">Mapping to Genomes with Multiple Contigs</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#overview-of-data-processing-procedure">Overview of Data Processing Procedure</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#statistics">Statistics</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Code Documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="transit.html">transit package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="transit.html#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit.html#pytransit-norm-tools-module">pytransit.norm_tools module</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit.html#pytransit-stat-tools-module">pytransit.stat_tools module</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit.html#pytransit-tnseq-tools-module">pytransit.tnseq_tools module</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit.html#pytransit-transit-tools-module">pytransit.transit_tools module</a></li>
<li class="toctree-l2"><a class="reference internal" href="transit.html#module-pytransit">Module contents</a></li>
</ul>
</li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">TRANSIT</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>Genetic Interactions</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/gi_method.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="genetic-interactions">
<span id="id1"></span><h1>Genetic Interactions<a class="headerlink" href="#genetic-interactions" title="Permalink to this headline">¶</a></h1>
<p>The genetic interactions (GI) method is a comparative analysis used
used to determine genetic interactions. It is a Bayesian method
that estimates the distribution of log fold-changes (logFC) in two
strain backgrounds under different conditions, and identifies significantly
large changes in enrichment (delta_logFC) to identify those genes
that imply a genetic interaction.</p>
<div class="line-block">
<div class="line"><br /></div>
</div>
<div class="section" id="how-does-it-work">
<h2>How does it work?<a class="headerlink" href="#how-does-it-work" title="Permalink to this headline">¶</a></h2>
<p>GI performs a comparison among 4 groups of datasets, strain A and B assessed in conditions 1 and 2 (e.g. control vs treatment).
It looks for interactions where the response to the treatment (i.e. effect on insertion counts) depends on the strain.</p>
<p>If you think of the effect of treatment as a log-fold-change (e.g. of
the insert counts between control and treatment in strain A), which is
like a “slope”, then the interacting genes are those that exhibit a difference
in the effect of the treatment between the strains, and hence a difference in the
slopes between strain A and B (represented by ‘delta_LFC’ in the output file).</p>
<div class="line-block">
<div class="line">For a formal description of how this method works, see our paper <a class="reference internal" href="transit_overview.html#dejesus2017nar" id="id2"><span>[DeJesus2017NAR]</span></a>:</div>
<div class="line"><br /></div>
<div class="line-block">
<div class="line">DeJesus, M.A., Nambi, S., Smith, C.M., Baker, R.E., Sassetti, C.M., Ioerger, T.R. <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/28334803">Statistical analysis of genetic interactions in Tn-Seq data.</a> <em>Nucleic Acids Research.</em> 2017. 45(11):e93. doi: 10.1093/nar/gkx128.</div>
</div>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
</div>
<div class="section" id="statistical-significance">
<h2>Statistical Significance<a class="headerlink" href="#statistical-significance" title="Permalink to this headline">¶</a></h2>
<p>The computation that is done by GI is to compute the posterior distribution of the delta_LFC (or mean change in slopes)
through Bayesian sampling.
The primary method to determine significance of genes is whethter this the mean_delta_LFC is significantly differnt than 0.
However, since the mean_delta_LFC is a distribution, we represent it by a Highest Density Interval, HDI, which is
similar to a 95% confidence interval.  Furthermore, rather than asking whether the HDI overlaps 0 exactly, we expand the interval
around 0 to a Region of Probable Equivalence (ROPE), which is set to [-0.5,0.5] by default.  Hence the significant genes
are those for which the HDI does not overlap the ROPE.  GI has a flag to  adjust the size of the ROPE, if desired.</p>
<p>In the GI output file, the final column give the significance call, along with type of interaction.
If a gene is not significant, it will be marked with “<strong>No Interaction</strong>” (for the HDI method, meaning HDI overlaps the ROPE).
If a gene IS significant, then its interaction will be cateogrized in 3 types (see NAR paper):</p>
<blockquote>
<div><ul class="simple">
<li><p><strong>Aggravating</strong> - mean_delta_LFC is negative; gene is more required in treatment than control in the B strain, compared to the A strain</p></li>
<li><p><strong>Suppressive</strong> - mean_delta_LFC is positive, and the gene was not conditionally essential in strain A (flat slope), but becomes conditionally non-essential in strain B when treated (positive slope)</p></li>
<li><p><strong>Alleviating</strong> - mean_delta_LFC is positive, but the conditional requirement (negative slope) of the gene in strain A with treatment is “cancelled” by the modification in strain B</p></li>
</ul>
</div></blockquote>
<img alt="_images/genetic_interaction_types.png" src="_images/genetic_interaction_types.png" />
<p>A limitation of this HDI approach is that it is discrete (i.e. overlap is either True or False), but does not provide a quantitative metric
for the degree of overlap.  Thus a second method for assessing significance of genetic interactions is to compute
the probability of overlap.  The lower the probability, the more differnt the delta_LFC is from 0, indicating a more
significant interaction.  In this case, genes with prob &lt; 0.05 are considered interactions and classified by the 3 types above,
while genes with prob &gt;= 0.05 are marked as “No Interaction”.</p>
<p>In addition, since we are calculating significance for thousands of genes in parallel,
many researchers prefer to have some method for correcting for multiple tests, to control the false discovery rate.
However, FDR correction is generally used only for frequentist analyses, and he GI method is fundamentally a Bayesian approach.
Technically, in a Bayesian framework, FDR correction is not needed.  Any adjustment for expectations about number of hits
would be achieved through adjusting parameters for prior distributions.  Nonetheless, GI includes options for
two methods that approximate FDR correction: <strong>BFDR</strong> (Bayesian False Discovery Rate correction,
<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/15054023/">Newton M.A., Noueiry A., Sarkar D., Ahlquist P. (2004). Detecting differential gene expression with a semiparametric hierarchical
mixture method. Biostatistics, 5:155–176.</a>) and FWER (Familty-Wise
Error Rate control).  When these corrections are applied, a threshold of 0.05 for the adjusted probability of overlap
is used for each, and this determines which
genes are classified as interacting (1 of 3 types) or  marked as “No Interaction”, as above.</p>
<p>In order to enable users to evaluate these various methods for determining significance of interactions,
a ‘-signif’ flag is provided for the GI method.  The options are:</p>
<blockquote>
<div><ul class="simple">
<li><p><strong>–signif HDI</strong>: significant genes are those for which the HDI does not overlap the ROPE</p></li>
<li><p><strong>–signif prob</strong>: significant genes are those with prob &lt; 0.05, where ‘prob’ is probability that HDI overlaps the ROPE (default)</p></li>
<li><p><strong>–signif BFDR</strong>: significant genes are those with adjusted prob &lt; 0.05, where prob is adjusted by the BFDR method</p></li>
<li><p><strong>–signif FWER</strong>: significant genes are those with adjusted prob &lt; 0.05, where prob is adjusted by the FWER method</p></li>
</ul>
</div></blockquote>
<p>‘–signif prob’ is the default method.</p>
<p>In the output file, the genes are sorted by the probability that the HDI overlaps the ROPE.
The genes at the top are rougly the genes with the highest absolute value of mean_delta_LFC.</p>
</div>
<div class="section" id="usage">
<h2>Usage<a class="headerlink" href="#usage" title="Permalink to this headline">¶</a></h2>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python3</span> <span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">GI</span> <span class="o">&lt;</span><span class="n">combined_wig_file</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">metadata_file</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">annotation_file</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">conditionA1</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">conditionB1</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">conditionA2</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">conditionB2</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">output_file</span><span class="o">&gt;</span> <span class="p">[</span><span class="n">optional</span> <span class="n">arguments</span><span class="p">]</span>

    <span class="n">Optional</span> <span class="n">Arguments</span><span class="p">:</span>
        <span class="o">--</span><span class="n">n</span> <span class="o">&lt;</span><span class="n">string</span><span class="o">&gt;</span>     <span class="o">:=</span>  <span class="n">Normalization</span> <span class="n">method</span><span class="o">.</span> <span class="n">Default</span><span class="p">:</span> <span class="o">--</span><span class="n">n</span> <span class="n">TTR</span>
        <span class="o">--</span><span class="n">s</span> <span class="o">&lt;</span><span class="n">integer</span><span class="o">&gt;</span>    <span class="o">:=</span>  <span class="n">Number</span> <span class="n">of</span> <span class="n">samples</span><span class="o">.</span> <span class="n">Default</span><span class="p">:</span> <span class="o">--</span><span class="n">s</span> <span class="mi">10000</span>
        <span class="o">--</span><span class="n">iN</span> <span class="o">&lt;</span><span class="nb">float</span><span class="o">&gt;</span>     <span class="o">:=</span>  <span class="n">Ignore</span> <span class="n">TAs</span> <span class="n">occurring</span> <span class="n">at</span> <span class="n">given</span> <span class="n">percentage</span> <span class="p">(</span><span class="k">as</span> <span class="n">integer</span><span class="p">)</span> <span class="n">of</span> <span class="n">the</span> <span class="n">N</span> <span class="n">terminus</span><span class="o">.</span> <span class="n">Default</span><span class="p">:</span> <span class="o">--</span><span class="n">iN</span> <span class="mi">0</span>
        <span class="o">--</span><span class="n">iC</span> <span class="o">&lt;</span><span class="nb">float</span><span class="o">&gt;</span>     <span class="o">:=</span>  <span class="n">Ignore</span> <span class="n">TAs</span> <span class="n">occurring</span> <span class="n">at</span> <span class="n">given</span> <span class="n">percentage</span> <span class="p">(</span><span class="k">as</span> <span class="n">integer</span><span class="p">)</span> <span class="n">of</span> <span class="n">the</span> <span class="n">C</span> <span class="n">terminus</span><span class="o">.</span> <span class="n">Default</span><span class="p">:</span> <span class="o">--</span><span class="n">iC</span> <span class="mi">0</span>
        <span class="o">--</span><span class="n">rope</span> <span class="o">&lt;</span><span class="nb">float</span><span class="o">&gt;</span>   <span class="o">:=</span>  <span class="n">Region</span> <span class="n">of</span> <span class="n">Practical</span> <span class="n">Equivalence</span><span class="o">.</span> <span class="n">Area</span> <span class="n">around</span> <span class="mi">0</span> <span class="p">(</span><span class="n">i</span><span class="o">.</span><span class="n">e</span><span class="o">.</span> <span class="mi">0</span> <span class="o">+/-</span> <span class="n">ROPE</span><span class="p">)</span> <span class="n">that</span> <span class="ow">is</span> <span class="n">NOT</span> <span class="n">of</span> <span class="n">interest</span><span class="o">.</span> <span class="n">Can</span> <span class="n">be</span> <span class="n">thought</span> <span class="n">of</span> <span class="n">similar</span> <span class="n">to</span> <span class="n">the</span> <span class="n">area</span> <span class="n">of</span> <span class="n">the</span> <span class="n">null</span><span class="o">-</span><span class="n">hypothesis</span><span class="o">.</span> <span class="n">Default</span><span class="p">:</span> <span class="o">--</span><span class="n">rope</span> <span class="mf">0.5</span>
        <span class="o">--</span><span class="n">signif</span> <span class="n">HDI</span>     <span class="o">:=</span>  <span class="p">(</span><span class="n">default</span><span class="p">)</span> <span class="n">Significant</span> <span class="k">if</span> <span class="n">HDI</span> <span class="n">does</span> <span class="ow">not</span> <span class="n">overlap</span> <span class="n">ROPE</span><span class="p">;</span> <span class="k">if</span> <span class="n">HDI</span> <span class="n">overlaps</span> <span class="n">ROPE</span><span class="p">,</span> <span class="s1">&#39;Type of Interaction&#39;</span> <span class="ow">is</span> <span class="nb">set</span> <span class="n">to</span> <span class="s1">&#39;No Interaction&#39;</span>
        <span class="o">--</span><span class="n">signif</span> <span class="n">prob</span>    <span class="o">:=</span>  <span class="n">Optionally</span><span class="p">,</span> <span class="n">significant</span> <span class="n">hits</span> <span class="n">are</span> <span class="n">re</span><span class="o">-</span><span class="n">defined</span> <span class="n">based</span> <span class="n">on</span> <span class="n">probability</span> <span class="p">(</span><span class="n">degree</span><span class="p">)</span> <span class="n">of</span> <span class="n">overlap</span> <span class="n">of</span> <span class="n">HDI</span> <span class="k">with</span> <span class="n">ROPE</span><span class="p">,</span> <span class="n">prob</span><span class="o">&lt;</span><span class="mf">0.05</span> <span class="p">(</span><span class="n">no</span> <span class="n">adjustment</span><span class="p">)</span>
        <span class="o">--</span><span class="n">signif</span> <span class="n">BFDR</span>    <span class="o">:=</span>  <span class="n">Apply</span> <span class="s2">&quot;Bayesian&quot;</span> <span class="n">FDR</span> <span class="n">correction</span> <span class="p">(</span><span class="n">see</span> <span class="n">doc</span><span class="p">)</span> <span class="n">to</span> <span class="n">adjust</span> <span class="n">HDI</span><span class="o">-</span><span class="n">ROPE</span> <span class="n">overlap</span> <span class="n">probabilities</span> <span class="n">so</span> <span class="n">that</span> <span class="n">significant</span> <span class="n">hits</span> <span class="n">are</span> <span class="n">re</span><span class="o">-</span><span class="n">defined</span> <span class="k">as</span> <span class="n">BFDR</span><span class="o">&lt;</span><span class="mf">0.05</span>
        <span class="o">--</span><span class="n">signif</span> <span class="n">FWER</span>    <span class="o">:=</span>  <span class="n">Apply</span> <span class="s2">&quot;Bayesian&quot;</span> <span class="n">FWER</span> <span class="n">correction</span> <span class="p">(</span><span class="n">see</span> <span class="n">doc</span><span class="p">)</span> <span class="n">to</span> <span class="n">adjust</span> <span class="n">HDI</span><span class="o">-</span><span class="n">ROPE</span> <span class="n">overlap</span> <span class="n">probabilities</span> <span class="n">so</span> <span class="n">that</span> <span class="n">significant</span> <span class="n">hits</span> <span class="n">are</span> <span class="n">re</span><span class="o">-</span><span class="n">defined</span> <span class="k">as</span> <span class="n">FWER</span><span class="o">&lt;</span><span class="mf">0.05</span>
</pre></div>
</div>
</div>
<div class="section" id="example">
<h2>Example<a class="headerlink" href="#example" title="Permalink to this headline">¶</a></h2>
<p>In this example, the effect of a knockout of SigB is being evaluated for its effect on tolerance of isoniazid.
Some genes may become more essential (or less) in the presence of INH in the wild-type strain.
The genes implied to interact with SigB are those whose response to INH changes in the knock-out strain compared to the wild-type.
Note the metadata file describes the grouping of multiple replicates in conditions
representing different combinations of strains and drug treatments (including WT_Untreated, SigB_Untreated, WT_INH, SigB_INH).</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python3</span> <span class="n">transit</span><span class="o">/</span><span class="n">src</span><span class="o">/</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">GI</span> <span class="n">antibiotics</span><span class="o">.</span><span class="n">combined_wig</span> <span class="n">antibiotics</span><span class="o">.</span><span class="n">metadata</span> <span class="n">mc2_155_tamu</span><span class="o">.</span><span class="n">prot_table</span> <span class="n">WT_Untreated</span> <span class="n">SigB_Untreated</span> <span class="n">WT_INH</span> <span class="n">SigB_INH</span> <span class="n">results</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<p>GI performs a comparison among 4 groups of datasets, strain A and B assessed in conditions 1 and 2 (e.g. control vs treatment).
It looks for interactions where the response to the treatment (i.e. effect on insertion counts) depends on the strain (output variable: delta_LFC).
Provide replicates in each group as a comma-separated list of wig files.
HDI is highest density interval for posterior distribution of delta_LFC, which is like a confidence interval on difference of slopes.
Genes are sorted by probability of HDI overlapping with ROPE. (genes with the highest abs(mean_delta_logFC) are near the top, approximately)
Significant genes are indicated by ‘Type of Interaction’ column (No Interaction, Aggravating, Alleviating, Suppressive).
By default, hits are defined as “Is HDI outside of ROPE?”=TRUE (i.e. non-overlap of delta_LFC posterior distritbuion with Region of Probably Equivalence around 0)
Alternative methods for significance: use -signif flag with prob, BFDR, or FWER. These affect ‘Type of Interaction’ (i.e. which genes are labeled ‘No Interaction’)</p>
</div>
<div class="section" id="parameters">
<h2>Parameters<a class="headerlink" href="#parameters" title="Permalink to this headline">¶</a></h2>
<p>The resampling method is non-parametric, and therefore does not require
any parameters governing the distributions or the model. The following
parameters are available for the method:</p>
<ul class="simple">
<li><p><strong>Samples:</strong> The number of samples (permutations) to perform. The
larger the number of samples, the more resolution the p-values
calculated will have, at the expense of longer computation time. The
resampling method runs on 10,000 samples by default.</p></li>
<li><p><strong>ROPE:</strong> Region of Practical Equivalence. This region defines an area
around 0.0 that represents differences in the log fold-change that are
practically equivalent to zero. This aids in ignoring spurious changes
in the logFC that would otherwise be identified under a strict
null-hypothesis of no difference.</p></li>
<li><p><strong>Include Zeros:</strong> Select to include  sites that are zero. This is the
preferred behavior, however, unselecting this (thus ignoring sites that)
are zero accross all dataset (i.e. completely empty), is useful for
decreasing running time (specially for large datasets like Tn5).</p></li>
<li><p><strong>Normalization Method:</strong> Determines which normalization method to
use when comparing datasets. Proper normalization is important as it
ensures that other sources of variability are not mistakenly treated
as real differences. See the <a class="reference internal" href="normalize_method.html#normalization"><span class="std std-ref">Normalization</span></a> section for a description
of normalization method available in TRANSIT.</p></li>
<li><p><strong>Significance Method:</strong></p></li>
</ul>
<blockquote>
<div><ul class="simple">
<li><p>-signif HDI: significant genes are those for which the HDI does not overlap the ROPE</p></li>
<li><p>-signif prob: significant genes are those with prob &lt; 0.05, where ‘prob’ is porbability that HDI overlap the ROPE (default)</p></li>
<li><p>-signif BFDR: significant genes are those with adjusted prob &lt; 0.05, where prob is adjusted by the BFDR method</p></li>
<li><p>-signif FWER: significant genes are those with adjusted prob &lt; 0.05, where prob is adjusted by the FWER method</p></li>
</ul>
</div></blockquote>
</div>
<div class="section" id="gui-mode">
<h2>Gui Mode<a class="headerlink" href="#gui-mode" title="Permalink to this headline">¶</a></h2>
<p>The Genetic Interactions analysis method can be selected from the “Method” tab in the Menu Bar.</p>
<a class="reference internal image-reference" href="_images/gi_selection_gui.png"><img alt="_images/gi_selection_gui.png" class="align-center" src="_images/gi_selection_gui.png" style="width: 1000px;" /></a>
<div class="line-block">
<div class="line"><br /></div>
</div>
<p>The parameters to input through the parameter panel for the method is equivalent to the command line usage (see parameter descriptions above for full detail):</p>
<a class="reference internal image-reference" href="_images/gi_parameter_panel.png"><img alt="_images/gi_parameter_panel.png" class="align-center" src="_images/gi_parameter_panel.png" style="width: 1000px;" /></a>
<p>The method is run using the combined wig, metadata, and annotation uploaded into TRANSIT.</p>
</div>
<div class="section" id="output-and-diagnostics">
<h2>Output and Diagnostics<a class="headerlink" href="#output-and-diagnostics" title="Permalink to this headline">¶</a></h2>
<p>The GI method outputs a tab-delimited file with results for each
gene in the genome.
All genes are sorted by significance using the probability that the HDI overlaps the ROPE.
Significant genes are those NOT marked with ‘No Interaction’ in the last column.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 40%" />
<col style="width: 60%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Column Header</p></th>
<th class="head"><p>Column Definition</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>ORF</p></td>
<td><p>Gene ID.</p></td>
</tr>
<tr class="row-odd"><td><p>Gene</p></td>
<td><p>Name of the gene.</p></td>
</tr>
<tr class="row-even"><td><p>Annotation</p></td>
<td><p>Description of gene per annotation_file</p></td>
</tr>
<tr class="row-odd"><td><p>TA Sites</p></td>
<td><p>Number of TA sites in the gene.</p></td>
</tr>
<tr class="row-even"><td><p>A1 Mean Count</p></td>
<td><p>Mean read count in strain A, condition 1</p></td>
</tr>
<tr class="row-odd"><td><p>A2 Mean Count</p></td>
<td><p>Mean read count in strain A, condition 2</p></td>
</tr>
<tr class="row-even"><td><p>B1 Mean Count</p></td>
<td><p>Mean read count in strain B, condition 1</p></td>
</tr>
<tr class="row-odd"><td><p>B2 Mean Count</p></td>
<td><p>Mean read count in strain B, condition 2</p></td>
</tr>
<tr class="row-even"><td><p>Log 2 FC Strain A</p></td>
<td><p>The log2 fold-change in read-count for strain A</p></td>
</tr>
<tr class="row-odd"><td><p>Log 2 FC Strain B</p></td>
<td><p>The log2 fold-change in read-count for strain B</p></td>
</tr>
<tr class="row-even"><td><p>Delta Log 2 FC</p></td>
<td><p>The difference in log2 fold-change between B and A</p></td>
</tr>
<tr class="row-odd"><td><p>Lower Bound Delta Log 2 FC</p></td>
<td><p>Lower bound of the difference (delta logFC)</p></td>
</tr>
<tr class="row-even"><td><p>Upper Bound Delta Log 2 FC</p></td>
<td><p>Upper bound of the difference (delta logFC)</p></td>
</tr>
<tr class="row-odd"><td><p>Is HDI Outside ROPE?</p></td>
<td><p>T/F value of whether Delta LFC distribution does not overlap 0</p></td>
</tr>
<tr class="row-even"><td><p>Probability of Delta Log 2 FC Being Within ROPE</p></td>
<td><p>Portion of the delta-logFC within ROPE</p></td>
</tr>
<tr class="row-odd"><td><p>Adjusted P Value</p></td>
<td><p>Posterior probability adjusted for comparisons using –signif method</p></td>
</tr>
<tr class="row-even"><td><p>Type of Interaction</p></td>
<td><p>Suppressive, Alleviating, Aggravating, No Interaction</p></td>
</tr>
</tbody>
</table>
<div class="line-block">
<div class="line"><br /></div>
</div>
<ul class="simple">
<li><p>If ‘–signif HDI’ is selected (default), then only genes where “Is HDI Outside ROPE?” is TRUE (delta LFC significantly different from 0) is categorized as an interaction (otherwise, ‘No Interaction’).</p></li>
<li><p>Interacting Genes can be <strong>ranked</strong> by strength of effect using the “Delta Log 2 FC” column.</p></li>
<li><p>Alternatively, users can sort by “Probability of Delta Log 2 FC Being Within ROPE”, where lower values indicate more significant interactions.</p></li>
<li><p>If other –signif options are selected, the genes categorized as interections depend on “Adjusted P Value”.</p></li>
</ul>
</div>
<div class="section" id="run-time">
<h2>Run-time<a class="headerlink" href="#run-time" title="Permalink to this headline">¶</a></h2>
<p>On an average machine, running this methodology on a dataset takes about ~15 minutes.</p>
<hr class="transit-sectionend docutils" />
</div>
</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="anova_method.html" class="btn btn-neutral float-right" title="ANOVA" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
        <a href="utest_method.html" class="btn btn-neutral float-left" title="Mann-Whitney U-test (utest)" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2015, Michael A. DeJesus &amp; Thomas R. Ioerger.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(false);
      });
  </script>

  
  
    
   

</body>
</html>