

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>CRISPRi-DR &mdash; TRANSIT v3.3.10 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/default.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> TRANSIT
          

          
            
            <img src="_static/transit_logo.png" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                v3.3.10
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption" role="heading"><span class="caption-text">TRANSIT MANUAL</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="transit_overview.html">TRANSIT Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_install.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_running.html">Running TRANSIT</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_quality_control.html">Quality Control</a></li>
<li class="toctree-l1"><a class="reference internal" href="file_formats.html">File Formats for Transit</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">TnSeq ANALYSIS METHODS</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="method_normalization.html">Normalization</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_tnseq_stats.html">tnseq_stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_gumbel.html">Gumbel</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_griffin.html">Griffin</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_ttnfitness.html">TTN-Fitness</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_tn5gaps.html">Tn5Gaps</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_HMM.html">HMM</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_resampling.html">Resampling</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_Utest.html">Mann-Whitney U-test (utest)</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_anova.html">ANOVA</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_corrplot.html">corrplot</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_heatmap.html">heatmap</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_GI.html">Genetic Interactions</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_zinb.html">ZINB</a></li>
<li class="toctree-l1"><a class="reference internal" href="method_pathway_enrichment.html">Pathway Enrichment Analysis</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">TRANSIT Tutorials</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="transit_interactions_tutorial.html">Tutorial: Genetic Interactions Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_normalization_tutorial.html">Tutorial: Normalize datasets</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_export_tutorial.html">Tutorial: Export datasets</a></li>
<li class="toctree-l1"><a class="reference internal" href="transit_console_cheatsheet.html">Console Mode Cheat-Sheet</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">TPP Manual</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="tpp.html">TPP Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#installation">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#running-tpp">Running TPP</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#mapping-to-genomes-with-multiple-contigs">Mapping to Genomes with Multiple Contigs</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#overview-of-data-processing-procedure">Overview of Data Processing Procedure</a></li>
<li class="toctree-l1"><a class="reference internal" href="tpp.html#statistics">Statistics</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Code Documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="transit.html">pytransit package</a></li>
<li class="toctree-l1"><a class="reference internal" href="example_scripts.html">Example Scripts (for Developers)</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">TRANSIT</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>CRISPRi-DR</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/CGI.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="transit-clionly section" id="crispri-dr">
<span id="cgi"></span><h1>CRISPRi-DR<a class="headerlink" href="#crispri-dr" title="Permalink to this headline">¶</a></h1>
<p>CRISPRi-DR is designed to analyze CRISPRi libraries from CGI experiments and identify significant CGIs ie genes that affect sensitivity to the drug when depleted.
[REF: TBA]</p>
<div class="section" id="workflow">
<h2>Workflow<a class="headerlink" href="#workflow" title="Permalink to this headline">¶</a></h2>
<p>Starting with fastq files, barcode counts are extracted. The user creates their own metadata file, for the counts. Fractional abundances are created using the counts files, the metadata file and the uninduced ATC counts file. The fractional abundances are then used to run the CRISPRi-DR model. The output of this model is a file that lists genes with their statistacal parameters and significance. Genes with significant interactions are those with <em>qval of condetration dependence &lt; 0.05</em> and <em>|Z score of concentration dependence|&gt;2</em>. However, genes can be ranked by depletion by sorting the coefficient of concentration dependence in ascending order</p>
<a class="reference internal image-reference" href="_images/CGI_workflow.png"><img alt="Alternative text" src="_images/CGI_workflow.png" style="width: 1000px;" /></a>
<p><strong>Preprocessing: Fastq to Count Files</strong></p>
<p>This is a longer process, taking a few minutes each. However, the number of reads processed is printed to the console to indicate progress.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../</span><span class="n">src</span><span class="o">/</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">extract_counts</span> <span class="o">&lt;</span><span class="n">fastq_file</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">ids_file</span><span class="o">&gt;</span> <span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">counts_file</span><span class="o">&gt;</span>
</pre></div>
</div>
<ul class="simple">
<li><dl class="simple">
<dt>ids_file<span class="classifier">List of sgRNAs used in the experiment, where each row is one sgRNA id.</span></dt><dd><ul>
<li><p>For H37Rv experiments, the ids file is available in : <code class="docutils literal notranslate"><span class="pre">transit/src/pytransit/data/CGI/IDs.H37Rv.CRISPRi.lib.txt</span></code></p></li>
</ul>
</dd>
</dl>
</li>
</ul>
<p><strong>Step 1: Combine Individual Counts File to a Combined Counts File</strong>
This is a fairly fast process. It takes at most a minute for the combination of 12 files with 2 columns (sgRNA id and counts) to one large file of 13 columns (first column sgRNA id and remaining columns are counts from the files entered).</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../</span><span class="n">src</span><span class="o">/</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">combined_counts</span> <span class="o">&lt;</span><span class="n">comma</span> <span class="n">seperated</span> <span class="n">headers</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">counts</span> <span class="n">file</span> <span class="mi">1</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">counts</span> <span class="n">file</span> <span class="mi">2</span><span class="o">&gt;</span>  <span class="o">...</span> <span class="o">&lt;</span><span class="n">counts_file</span> <span class="n">n</span><span class="o">&gt;</span> <span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">combined</span> <span class="n">counts</span> <span class="n">file</span><span class="o">&gt;</span>
</pre></div>
</div>
<ul>
<li><p>counts files : sgRNA ids as their first column, and can have any number of columns.</p></li>
<li><dl>
<dt>comma-separated headers: the column names of the combined counts file</dt><dd><div class="admonition note">
<p class="admonition-title">Note</p>
<p>the comma-seperated headers must be in order of the columns in the count files(s) provided</p>
</div>
</dd>
</dl>
</li>
</ul>
<p><strong>Step 2: Extract Fractional Abundances</strong></p>
<blockquote>
<div><p>This is a relatively quick process, taking less than a minute. This step is to turn the barcodes counts into relative normalized abundances. Counts are normalized within samples and calculated relative to the abundances in the uninduced ATC file, essentially fractions.</p>
<p>Each mutant has an sgRNA mapping to a target gene that can reduce its expression (when induced with ATC, anhydrotetracycline). The uninduced ATC counts files are the read counts when the sgRNAs are mapped to a target gene but have not been induced. The uninduced counts provide a starting point to compare with the ATC induced zero concentration counts as well as the ATC inducted non-zero concentration counts.</p>
<p>The first few lines of the output file contains information about the counts files processed.</p>
</div></blockquote>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../</span><span class="n">src</span><span class="o">/</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">extract_abund</span> <span class="o">&lt;</span><span class="n">combined</span> <span class="n">counts</span> <span class="n">file</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">counts</span> <span class="n">metadata</span> <span class="n">file</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">control</span> <span class="n">condition</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">sgRNA</span> <span class="n">efficiencies</span> <span class="n">file</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">uninduced</span> <span class="n">ATC</span> <span class="n">file</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">drug</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">days</span><span class="o">&gt;</span>  <span class="o">&gt;</span>  <span class="o">&lt;</span><span class="n">fractional</span> <span class="n">abundance</span> <span class="n">file</span><span class="o">&gt;</span>
</pre></div>
</div>
<ul>
<li><p>counts metadata file (USER created):</p>
<blockquote>
<div><ul>
<li><p>The columns expected in this file: column_name, drug, conc_xMIC, days_predepletion</p>
<blockquote>
<div><ul class="simple">
<li><p>column_name: the corresponding header name(s) in the combined counts file</p></li>
<li><p>conc_xMIC is the concentration of the drug the sample is treated with</p></li>
</ul>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>conc_xMIC must be a numerical value, ie. 0.5 and not a categorical value such as “low” or “high”</p>
</div>
<ul class="simple">
<li><p>Equal number of replicates for all concentrations are not nessessary</p></li>
<li><p>see [Li, S et al. 2022, PMID: 35637331] for explanation of days_predepletion</p></li>
</ul>
</div></blockquote>
</li>
<li><p>Example metadata: <code class="docutils literal notranslate"><span class="pre">transit/src/pytransit/data/CGI/samples_metadata.txt</span></code></p></li>
</ul>
</div></blockquote>
</li>
<li><p>control condition: The condition to to be considered the control for these set of experiments, as specificed in the “drug” column of the metadata file; typically an atc-induced (+ ATC) with 0 drug concentration condition.</p></li>
<li><p>sgRNA efficiencies file: A file that contains metadata for each sgRNA in the combined counts file, where the first column must be sgRNA id (as seen in the combined counts file) and the last column must be the efficiency measurement of the sgRNAs (in publication of this method, sgRNA efficiency is measurement as extrapolated LFCs calculated through a passaging experiment).</p></li>
<li><p>uninduced ATC file: A two column file of sgRNAs and their counts in uninduced ATC (no ATC) with 0 drug concentration</p></li>
<li><p>drug : Name of the drug in the “drug” column of the metadata file passed in to be fit in the model</p></li>
<li><p>days: Sampled from predepletion day as listed in the “days_predepletion” column of the metadata file to be used in the analysis</p></li>
</ul>
<p><strong>Step 3: Run the CRISPRi-DR model</strong></p>
<p>This is a relatively quick process, taking at most 3 minutes for a dataset of ~90,000 sgRNAs . This step fits the CRISPRi-DR model (statistical analysis of concentration dependence for each gene) to each gene in the file and prints each output to the &lt;CRISPRi-DR results file&gt; in a tab seperated file.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../</span><span class="n">src</span><span class="o">/</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">run_model</span> <span class="o">&lt;</span><span class="n">fractional</span> <span class="n">abundance</span> <span class="n">file</span><span class="o">&gt;</span>  <span class="o">&gt;</span>  <span class="o">&lt;</span><span class="n">CRISPRi</span><span class="o">-</span><span class="n">DR</span> <span class="n">results</span> <span class="n">file</span><span class="o">&gt;</span>
</pre></div>
</div>
<p>The output columns in this file are:</p>
<ul class="simple">
<li><p>Significant Interactions - Siginificant interacting genes are those with <em>qval of concentration dependence &lt; 0.05</em> and <em>|Z score of concentration dependence|&gt;2</em>, these are indicated by a “-1” for depleted and “1” for enriched in in the “Significant Interactions” column</p></li>
<li><p>Orf - the orf id of the gene</p></li>
<li><p>Gene - the name of the gene</p></li>
<li><p>Nobs - the number of sgRNAs seen in an orf</p></li>
<li><p>intercept - the resulting intercept of the overall fitted regression</p></li>
<li><p>coefficient sgRNA_efficiency - coefficient of the amount sgRNA efficiency contributes to the decrease in abundance</p></li>
<li><p>coefficient concentration dependence - coefficient of the amount concentration contributes to the decrease in abundance</p></li>
<li><p>pval intercept - the wald test based p-value of the intercept</p></li>
<li><p>pval sgRNA_efficiency - the wald test based p-value of the coefficient sgRNA_efficiency</p></li>
<li><p>pval concentration dependence - the wald test based p-value of the coefficient of concentration dependence</p></li>
<li><p>qval concentration dependence - FDR corrected adjusted pvalues of of the coefficient of concentration dependence</p></li>
<li><p>Z score of concentration dependence - zscore of the coefficient of concentration dependence</p></li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>When the file is sorted on the slope of concentration dependence, the user can rank the genes based on amount of depletion.</p>
</div>
<p><strong>Visualize Specific Genes</strong></p>
<p>This process is fairly quick, taking less than a minute to run. This figure visualizes the amount of depletion in a gene at the sgRNA level. If control concentration provided is 0, the lowest value on the x-axis in the plot refers to this concentration (due to taking log concentration, 0 concentration is treated as a teo fold lower than the lowest concentration.) The slope of relative abundance (fraction of abundance of counts in ATC induced vs. ATC uninduced) versus log2(concentration) for each sgRNA is calculated and plotted, colored by sgRNA strenght based on a blue-orange gradient (as seen here):</p>
<a class="reference internal image-reference" href="_images/RVBD3645_lmplot.png"><img alt="Alternative text" src="_images/RVBD3645_lmplot.png" style="width: 400px;" /></a>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../</span><span class="n">src</span><span class="o">/</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">visualize</span> <span class="o">&lt;</span><span class="n">fractional</span> <span class="n">abundance</span> <span class="n">file</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">gene</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">output</span> <span class="n">plot</span> <span class="n">location</span><span class="o">&gt;</span>
</pre></div>
</div>
<ul>
<li><p>fractional abundance file : Fractional abundance file as created in Step 2.</p>
<blockquote>
<div><div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>This visualization assumes the columns are in increasing order of concentration, with the first three abundance columns (after the column “sgRNA efficiency”), as the control. This order depends on the order of columns during the creation of the combined counts file in Step 1.</p>
</div>
</div></blockquote>
</li>
<li><p>gene : select a gene to visualize. Use orf or gene name</p></li>
<li><p>output plot location : The location where to save the generated plot.</p></li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If comparing plots from different genes, note the scale of sgRNA efficiency shown in the plots.</p>
</div>
</div>
<div class="section" id="tutorial">
<h2>Tutorial<a class="headerlink" href="#tutorial" title="Permalink to this headline">¶</a></h2>
<p>This tutorial shows commands relative to this directory. Files in the <code class="docutils literal notranslate"><span class="pre">transit/src/pytransit/data/CGI</span></code> directory are:</p>
<ul class="simple">
<li><p>samples_metadata.txt - describes the samples</p></li>
<li><p>sgRNA_info.txt - contains extrapolated LFCs for each sgRNA</p></li>
<li><p>uninduced_ATC_counts.txt - counts for uninduced ATC (no induction of target depletion) library</p></li>
<li><p>IDs.H37Rv.CRISPRi.lib.txt - ids of the sgRNAs that target the genes in H37Rv used in these experiments</p></li>
<li><p>RIF_D1_combined_counts.txt - combined counts of the RIF 1 day predepletion data for uninduced ATC, zero, low, medium and high concentrations (output of data preprocessed and Step 1 completed)</p></li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If the user would like to evaluate the software, they can start with Step 2, using the <em>RIF_D1_combined_counts.txt</em> file in the <code class="docutils literal notranslate"><span class="pre">transit/src/pytransit/data/CGI</span></code> directory.</p>
</div>
<p><strong>Raw Data : Obtain FastQ files from NCBI using the following run numbers</strong></p>
<p>Fetch and process the following into fastQ files from <a class="reference external" href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA738381/">NCBI</a> using the SRA toolkit and place them in the <code class="docutils literal notranslate"><span class="pre">transit/src/pytransit/data/CGI</span></code> directory :</p>
<ul class="simple">
<li><p>Control samples (ATC-induced 0 drug concentration DMSO library with 1 day predepletion) : SRR14827863, SRR14827862, SRR14827799</p></li>
<li><p>High concentration RIF in a 1 day pre-depletion library : SRR14827727, SRR14827861, SRR14827850</p></li>
<li><p>Medium concentration RIF in a 1 day pre-depletion library: SRR14827760, SRR14827749, SRR14827738</p></li>
<li><p>Low concentration RIF in a 1 day pre-depletion library: SRR14827769, SRR14827614, SRR14827870</p></li>
</ul>
<p><strong>Preprocessing: Fastq to Count Files</strong></p>
<p>Create file of barcode counts from fastq files. Each fastq files reflect one replicate of a drug concentration, thus each will be converted into a file with two columns, sgNRA id and barcode counts</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">extract_counts</span> <span class="n">RIF_fastq_files</span><span class="o">/</span><span class="n">SRR14827863_1</span><span class="o">.</span><span class="n">fastq</span> <span class="n">IDs</span><span class="o">.</span><span class="n">H37Rv</span><span class="o">.</span><span class="n">CRISPRi</span><span class="o">.</span><span class="n">lib</span><span class="o">.</span><span class="n">txt</span> <span class="o">&gt;</span> <span class="n">DMSO_D1_rep1</span><span class="o">.</span><span class="n">counts</span>
<span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">extract_counts</span> <span class="n">RIF_fastq_files</span><span class="o">/</span><span class="n">SRR14827862_1</span><span class="o">.</span><span class="n">fastq</span> <span class="n">IDs</span><span class="o">.</span><span class="n">H37Rv</span><span class="o">.</span><span class="n">CRISPRi</span><span class="o">.</span><span class="n">lib</span><span class="o">.</span><span class="n">txt</span> <span class="o">&gt;</span> <span class="n">DMSO_D1_rep2</span><span class="o">.</span><span class="n">counts</span>
<span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">extract_counts</span> <span class="n">RIF_fastq_files</span><span class="o">/</span><span class="n">SRR14827799_1</span><span class="o">.</span><span class="n">fastq</span> <span class="n">IDs</span><span class="o">.</span><span class="n">H37Rv</span><span class="o">.</span><span class="n">CRISPRi</span><span class="o">.</span><span class="n">lib</span><span class="o">.</span><span class="n">txt</span> <span class="o">&gt;</span> <span class="n">DMSO_D1_rep3</span><span class="o">.</span><span class="n">counts</span>

<span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">extract_counts</span> <span class="n">RIF_fastq_files</span><span class="o">/</span><span class="n">SRR14827769_1</span><span class="o">.</span><span class="n">fastq</span> <span class="n">IDs</span><span class="o">.</span><span class="n">H37Rv</span><span class="o">.</span><span class="n">CRISPRi</span><span class="o">.</span><span class="n">lib</span><span class="o">.</span><span class="n">txt</span> <span class="o">&gt;</span> <span class="n">RIF_D1_Low_rep1</span><span class="o">.</span><span class="n">counts</span>
<span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">extract_counts</span> <span class="n">RIF_fastq_files</span><span class="o">/</span><span class="n">SRR14827614_1</span><span class="o">.</span><span class="n">fastq</span> <span class="n">IDs</span><span class="o">.</span><span class="n">H37Rv</span><span class="o">.</span><span class="n">CRISPRi</span><span class="o">.</span><span class="n">lib</span><span class="o">.</span><span class="n">txt</span> <span class="o">&gt;</span> <span class="n">RIF_D1_Low_rep2</span><span class="o">.</span><span class="n">counts</span>
<span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">extract_counts</span> <span class="n">RIF_fastq_files</span><span class="o">/</span><span class="n">SRR14827870_1</span><span class="o">.</span><span class="n">fastq</span> <span class="n">IDs</span><span class="o">.</span><span class="n">H37Rv</span><span class="o">.</span><span class="n">CRISPRi</span><span class="o">.</span><span class="n">lib</span><span class="o">.</span><span class="n">txt</span> <span class="o">&gt;</span> <span class="n">RIF_D1_Low_rep3</span><span class="o">.</span><span class="n">counts</span>

<span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">extract_counts</span> <span class="n">RIF_fastq_files</span><span class="o">/</span><span class="n">SRR14827760_1</span><span class="o">.</span><span class="n">fastq</span> <span class="n">IDs</span><span class="o">.</span><span class="n">H37Rv</span><span class="o">.</span><span class="n">CRISPRi</span><span class="o">.</span><span class="n">lib</span><span class="o">.</span><span class="n">txt</span> <span class="o">&gt;</span> <span class="n">RIF_D1_Med_rep1</span><span class="o">.</span><span class="n">counts</span>
<span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">extract_counts</span> <span class="n">RIF_fastq_files</span><span class="o">/</span><span class="n">SRR14827749_1</span><span class="o">.</span><span class="n">fastq</span> <span class="n">IDs</span><span class="o">.</span><span class="n">H37Rv</span><span class="o">.</span><span class="n">CRISPRi</span><span class="o">.</span><span class="n">lib</span><span class="o">.</span><span class="n">txt</span> <span class="o">&gt;</span> <span class="n">RIF_D1_Med_rep2</span><span class="o">.</span><span class="n">counts</span>
<span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">extract_counts</span> <span class="n">RIF_fastq_files</span><span class="o">/</span><span class="n">SRR14827738_1</span><span class="o">.</span><span class="n">fastq</span> <span class="n">IDs</span><span class="o">.</span><span class="n">H37Rv</span><span class="o">.</span><span class="n">CRISPRi</span><span class="o">.</span><span class="n">lib</span><span class="o">.</span><span class="n">txt</span> <span class="o">&gt;</span> <span class="n">RIF_D1_Med_rep3</span><span class="o">.</span><span class="n">counts</span>

<span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">extract_counts</span> <span class="n">RIF_fastq_files</span><span class="o">/</span><span class="n">SRR14827727_1</span><span class="o">.</span><span class="n">fastq</span> <span class="n">IDs</span><span class="o">.</span><span class="n">H37Rv</span><span class="o">.</span><span class="n">CRISPRi</span><span class="o">.</span><span class="n">lib</span><span class="o">.</span><span class="n">txt</span> <span class="o">&gt;</span> <span class="n">RIF_D1_High_rep1</span><span class="o">.</span><span class="n">counts</span>
<span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">extract_counts</span> <span class="n">RIF_fastq_files</span><span class="o">/</span><span class="n">SRR14827861_1</span><span class="o">.</span><span class="n">fastq</span> <span class="n">IDs</span><span class="o">.</span><span class="n">H37Rv</span><span class="o">.</span><span class="n">CRISPRi</span><span class="o">.</span><span class="n">lib</span><span class="o">.</span><span class="n">txt</span> <span class="o">&gt;</span> <span class="n">RIF_D1_High_rep2</span><span class="o">.</span><span class="n">counts</span>
<span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">extract_counts</span> <span class="n">RIF_fastq_files</span><span class="o">/</span><span class="n">SRR14827850_1</span><span class="o">.</span><span class="n">fastq</span> <span class="n">IDs</span><span class="o">.</span><span class="n">H37Rv</span><span class="o">.</span><span class="n">CRISPRi</span><span class="o">.</span><span class="n">lib</span><span class="o">.</span><span class="n">txt</span> <span class="o">&gt;</span> <span class="n">RIF_D1_High_rep3</span><span class="o">.</span><span class="n">counts</span>
</pre></div>
</div>
<p><strong>Step 1: Combine Counts Files to a Combined Counts File</strong></p>
<p>Combine the 12 seperate counts files into one combined counts file. Here we put the control samples first (DMSO) and then the drug-treated libraries (RIF) in increasing concentration</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">create_combined_counts</span> <span class="n">DMSO_D1_rep1</span><span class="p">,</span><span class="n">DMSO_D1_rep2</span><span class="p">,</span><span class="n">DMSO_D1_rep3</span><span class="p">,</span><span class="n">RIF_D1_Low_rep1</span><span class="p">,</span><span class="n">RIF_D1_Low_rep2</span><span class="p">,</span><span class="n">RIF_D1_Low_rep3</span><span class="p">,</span><span class="n">RIF_D1_Med_rep1</span><span class="p">,</span><span class="n">RIF_D1_Med_rep2</span><span class="p">,</span><span class="n">RIF_D1_Med_rep3</span><span class="p">,</span><span class="n">RIF_D1_High_rep1</span><span class="p">,</span><span class="n">RIF_D1_High_rep2</span><span class="p">,</span><span class="n">RIF_D1_High_rep3</span>  <span class="n">DMSO_D1_rep1</span><span class="o">.</span><span class="n">counts</span> <span class="n">DMSO_D1_rep2</span><span class="o">.</span><span class="n">counts</span> <span class="n">DMSO_D1_rep3</span><span class="o">.</span><span class="n">counts</span> <span class="n">RIF_D1_Low_rep1</span><span class="o">.</span><span class="n">counts</span> <span class="n">RIF_D1_Low_rep2</span><span class="o">.</span><span class="n">counts</span> <span class="n">RIF_D1_Low_rep3</span><span class="o">.</span><span class="n">counts</span> <span class="n">RIF_D1_Med_rep1</span><span class="o">.</span><span class="n">counts</span> <span class="n">RIF_D1_Med_rep2</span><span class="o">.</span><span class="n">counts</span> <span class="n">RIF_D1_Med_rep3</span><span class="o">.</span><span class="n">counts</span> <span class="n">RIF_D1_High_rep1</span><span class="o">.</span><span class="n">counts</span> <span class="n">RIF_D1_High_rep2</span><span class="o">.</span><span class="n">counts</span> <span class="n">RIF_D1_High_rep3</span><span class="o">.</span><span class="n">counts</span> <span class="o">&gt;</span> <span class="n">RIF_D1_combined_counts</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<p>The resulting file will have 13 columns, where the first column is sgRNA ids and the remaining are the counts for three replicates each for DMSO, RIF D1 Low Concentration, RIF D1 Med Concentration and RIF D1 High Concentration, respectively.</p>
<p><strong>Step 2: Extract Fractional Abundances</strong></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>As a part of this step, the <em>user must also generate a metadata file.</em> , ie. <code class="docutils literal notranslate"><span class="pre">samples_metadata.txt</span></code>. Note the values in the conc_xMIC column is actual values (0.0625, 0.125, 0.25) and not categorical values (“low”, “medium”, “high”) as seen in the counts file names.</p>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">extract_abund</span> <span class="n">RIF_D1_combined_counts</span><span class="o">.</span><span class="n">txt</span> <span class="n">samples_metadata</span><span class="o">.</span><span class="n">txt</span> <span class="n">DMSO</span> <span class="n">sgRNA_info</span><span class="o">.</span><span class="n">txt</span> <span class="n">uninduced_ATC_counts</span><span class="o">.</span><span class="n">txt</span> <span class="n">RIF</span> <span class="mi">1</span>  <span class="o">&gt;</span>  <span class="n">RIF_D1_frac_abund</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<p>The result of this command should be a file with a set of comments at the top, detailing the libraries used (DMSO and RIF). There should be a total of 17 columns, the last 12 of which are the calculated abundances, the first is the sgRNA ids followed by the orf/gene the sgRNA is targeting, uninduced ATC values, and sgRNA efficiency.</p>
<p><strong>Step 3: Run the CRISPRi-DR model</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">run_model</span> <span class="n">RIF_D1_frac_abund</span><span class="o">.</span><span class="n">txt</span> <span class="o">&gt;</span> <span class="n">RIF_D1_CRISPRi</span><span class="o">-</span><span class="n">DR_results</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<p>There should be a total of 184 significant gene interactions, where 111 are significant depletions and 73 are significantly enriched.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>When the file is sorted on the slope of concentration dependence, the user can rank the genes based on amount of depletion.</p>
</div>
<p><strong>Visualize Specific Genes</strong></p>
<p>Here are a few samples of the interactions visualized at the sgRNA level for this experiment. Note the difference in sgRNA efficiency scales shown.</p>
<p><em>Significantly depleted gene : RVBD3645</em></p>
<p><em>RVBD3645</em> is one of the significantly depleted genes in this experiment. In this plot, notice how most of the slopes are negative but the amount of depletion varies, where the more blue slopes (higher sgRNA efficiency) are steeper than orange sgRNA slopes (lower sgRNA efficiency)</p>
<a class="reference internal image-reference" href="_images/RVBD3645_lmplot.png"><img alt="Alternative text" src="_images/RVBD3645_lmplot.png" style="width: 400px;" /></a>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">visualize</span> <span class="n">RIF_D1_frac_abund</span><span class="o">.</span><span class="n">txt</span> <span class="n">RVBD3645</span> <span class="o">./</span><span class="n">RVBD3645_lmplot</span><span class="o">.</span><span class="n">png</span>
</pre></div>
</div>
<p><em>Significantly enriched gene : ndh</em></p>
<p><em>ndh</em> is one of the signifincantly enriched genes in this experiment. In its plot, notice how sgRNAs of high efficiency (blue green ones) show a strong upwards trend but those will lower efficiency (the orange ones) do not. In fact there a few sgRNAs that show almost no change in fractional abundace as concentration increases.</p>
<a class="reference internal image-reference" href="_images/ndh_lmplot.png"><img alt="Alternative text" src="_images/ndh_lmplot.png" style="width: 400px;" /></a>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">visualize</span> <span class="n">RIF_D1_frac_abund</span><span class="o">.</span><span class="n">txt</span> <span class="n">ndh</span> <span class="o">./</span><span class="n">ndh_lmplot</span><span class="o">.</span><span class="n">png</span> <span class="c1">#enriched</span>
</pre></div>
</div>
<p><em>Non-interacting gene : thiL</em></p>
<p><em>thiL</em> is an example on an non-interacting gene. It was found to be neither signifinicantly enriched nor depleted. Notice how in its plot, most of the slopes are fairly flat. As seen in the plots of <em>RVBD3645</em> and <em>ndh</em>, the bluer slopes show greater depletion than the orange slopes, but there is no overall trend present</p>
<a class="reference internal image-reference" href="_images/thiL_lmplot.png"><img alt="Alternative text" src="_images/thiL_lmplot.png" style="width: 400px;" /></a>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="n">python3</span> <span class="o">../../../</span><span class="n">transit</span><span class="o">.</span><span class="n">py</span> <span class="n">CGI</span> <span class="n">visualize</span> <span class="n">RIF_D1_frac_abund</span><span class="o">.</span><span class="n">txt</span> <span class="n">thiL</span> <span class="o">./</span><span class="n">thiL_lmplot</span><span class="o">.</span><span class="n">png</span>
</pre></div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2015, Michael A. DeJesus &amp; Thomas R. Ioerger.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>